---
title: "Fungicide - Differential Abundance & OccupancyAbundance"
author: "Zachary Noel"
date: "5/23/2020"
output: html_document
---

## Loading packages
```{r SOURCE}
source('functions_themes.R')
```

```{r setup, include=FALSE}
#source('http://bioconductor.org/biocLite.R')
#biocLite('phyloseq')
library(phyloseq)
library(decontam)
#browseVignettes("phyloseq")

packages <- c("biomformat", "qiimer", "MASS", "ape", "ggplot2", "plyr", "indicspecies", "labdsv", "dplyr", "reshape", "vegan", "metacoder", "lme4", "lsmeans", "ggpmisc", "ggpubr", "tidyverse", "doParallel", "DT", "exactRankTests", "foreach", "Rcpp", "shiny", "coin", "igraph", "SpiecEasi", "ggsci", "RCy3", "ggrepel", "cowplot", "ggalt", "ggfortify", "emmeans", "data.table", "randomForest", "rfUtilities", "caret", "tidyr")
ipak(packages)

packageVersion("vegan")
packageVersion('phyloseq')

#install.packages("remotes")
#remotes::install_github("DanielSprockett/reltools")
library(reltools)
#install.packages("minpack.lm")
library(minpack.lm)
#Models for the whole community
#install.packages("devtools")
library(devtools)
#install_github("DanielSprockett/tyRa")
#install.packages("Hmisc")
library(Hmisc)
library(Biostrings)
library(ggforce)
library(cowplot)
library(randomForest)
library(rfUtilities) # to test model significance
library(caret) # to get leave-one-out cross-validation accuracies and also contains the nearZeroVar function 
library(NetworkExtinction)
library(ggraph)
library(tidygraph)
```

## FUNGI CSS NORM RDS
Save the fungi phyloseq object as an RDS file to load faster in future.
```{r FUNGI CSS NORM - RDS}
fungi.CSS.norm.filt <- readRDS(file = "Preprocessing/Fungicide_Fungi_RootsAndLeaves_1000seqMin_CSSNorm_022421.rds")
fungi.no.norm.filt <- readRDS(file = "Preprocessing/Fungicide_Fungi_RootsAndLeaves_1000readsmin_nonorm_022421.rds")
```

## PROKARYOTE CSS NORM and Non-normalized RDS
```{r}
# Restore the object
bacteria.no.norm <- readRDS("Preprocessing/Fungicide_Bacteria_RootsAndLeaves_newtaxa_nonorm_010621.rds")
bacteria.css.norm <- readRDS(file = "Preprocessing/Fungicide_Bacteria_RootsAndLeaves_newtaxa_CSSnorm_010621.rds")
```

### Taxonomy of prokaryotes and fungi
```{r}
taxonomy <- fungi.no.norm.filt@tax_table %>%
  as("matrix") %>%
  replace_na("unidentified") %>%
  as_tibble() %>%
  mutate(OTU = OTU_ID)

taxonomy.prok <- bacteria.no.norm@tax_table %>%
  as("matrix") %>%
  replace_na("unidentified") %>%
  as_tibble() %>%
  mutate(OTU = OTU_ID)
```


### ANCOM - differentially abundant OTUs
-There was evidence that corn phyllosphere was altered dependent on fungicide and the when the fungicide was sprayed.
-False positives were controlled statistically within ANCOM with a bonferonii pvalue adjustment. 

-Input files into ANCOM, going to filter OTUs that have less than a mean relative abundance less than 1e-5
-Input into ANCOM is raw count data since it performs its own normalization based on CLR transformation

-We will test for differentially abundant taxa on soybean at growthstage R4 (9 dpi) and R6 (33 dpi) since composition was altered then
-For corn, we will test for deferentially abundant taxa at growthstage V8 only since there was no evidence that composition as altered past one week 
-ANCOM contains a filter to take out OTUs with a zero count in greater tahn 95% samples (i.e., occupancy greater than 5%)
-We will also test each management scheme separately meaning, our n = 12, 12 samples for Fungicide compared to 12 samples non-sprayed


- YOU CAN SKIP THIS CHUNK IF YOU HAVE THE RESULT RDS FILES
```{r}
#### CORN V8 T1####
fungi.keep.corn.v8.t1.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Corn" & GrowthStage == "V8" & Treatment == "T1" & Compartment == "Leaf") %>%
  phyloseq::transform_sample_counts(function(x) x / sum(x) ) %>%
  phyloseq::filter_taxa(function(x) mean(x) > 1e-5, TRUE) %>%
  taxa_names()

corn.v8.t1.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Corn" & GrowthStage == "V8" & Treatment == "T1" & Compartment == "Leaf")
  
fungi_leaf_corn_V8_T1_filt <- prune_taxa(fungi.keep.corn.v8.t1.leaf, corn.v8.t1.leaf) # remove the taxa below 1e-5

#### CORN V8 T2####

fungi.keep.corn.v8.t2.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Corn" & GrowthStage == "V8" & Treatment == "T2" & Compartment == "Leaf") %>%
  phyloseq::transform_sample_counts(function(x) x / sum(x) ) %>%
  phyloseq::filter_taxa(function(x) mean(x) > 1e-5, TRUE) %>%
  taxa_names()

corn.v8.t2.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Corn" & GrowthStage == "V8" & Treatment == "T2" & Compartment == "Leaf")
  
fungi_leaf_corn_V8_T2_filt <- prune_taxa(fungi.keep.corn.v8.t2.leaf, corn.v8.t2.leaf) # remove the taxa below 1e-5

#### SOY R4 T1####

fungi.keep.soy.R4.T1.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R4" & Treatment == "T1" & Compartment == "Leaf") %>%
  phyloseq::transform_sample_counts(function(x) x / sum(x) ) %>%
  phyloseq::filter_taxa(function(x) mean(x) > 1e-5, TRUE) %>%
  taxa_names()

soy.R4.T1.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R4" & Treatment == "T1" & Compartment == "Leaf")
  
fungi_leaf_soy_R4_T1_filt <- prune_taxa(fungi.keep.soy.R4.T1.leaf, soy.R4.T1.leaf) # remove the taxa below 1e-5

#### SOY R4 T2####

fungi.keep.soy.R4.T2.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R4" & Treatment == "T2" & Compartment == "Leaf") %>%
  phyloseq::transform_sample_counts(function(x) x / sum(x) ) %>%
  phyloseq::filter_taxa(function(x) mean(x) > 1e-5, TRUE) %>%
  taxa_names()

soy.R4.T2.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R4" & Treatment == "T2" & Compartment == "Leaf")
  
fungi_leaf_soy_R4_T2_filt <- prune_taxa(fungi.keep.soy.R4.T2.leaf, soy.R4.T2.leaf) # remove the taxa below 1e-5

#### SOY R6 T1####

fungi.keep.soy.R6.T1.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R6" & Treatment == "T1" & Compartment == "Leaf") %>%
  phyloseq::transform_sample_counts(function(x) x / sum(x) ) %>%
  phyloseq::filter_taxa(function(x) mean(x) > 1e-5, TRUE) %>%
  taxa_names()

soy.R6.T1.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R6" & Treatment == "T1" & Compartment == "Leaf")
  
fungi_leaf_soy_R6_T1_filt <- prune_taxa(fungi.keep.soy.R6.T1.leaf, soy.R6.T1.leaf) # remove the taxa below 1e-5

#### SOY R6 T2####

fungi.keep.soy.R6.T2.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R6" & Treatment == "T2" & Compartment == "Leaf") %>%
  phyloseq::transform_sample_counts(function(x) x / sum(x) ) %>%
  phyloseq::filter_taxa(function(x) mean(x) > 1e-5, TRUE) %>%
  taxa_names()

soy.R6.T2.leaf <- fungi.no.norm.filt %>%
  subset_samples(Crop == "Soy" & GrowthStage == "R6" & Treatment == "T2" & Compartment == "Leaf")
  
fungi_leaf_soy_R6_T2_filt <- prune_taxa(fungi.keep.soy.R6.T2.leaf, soy.R6.T2.leaf) # remove the taxa below 1e-5


#### SAVE RDS FILES FOR HPCC ####
saveRDS(fungi_leaf_corn_V8_T1_filt, "ANCOM/Input/fungi_leaf_corn_V8_T1_filt_022621.rds")
saveRDS(fungi_leaf_corn_V8_T2_filt, "ANCOM/Input/fungi_leaf_corn_V8_T2_filt_022621.rds")
saveRDS(fungi_leaf_soy_R4_T1_filt, "ANCOM/Input/fungi_leaf_soy_R4_T1_filt_022621.rds")
saveRDS(fungi_leaf_soy_R6_T1_filt, "ANCOM/Input/fungi_leaf_soy_R6_T1_filt_022621.rds")
saveRDS(fungi_leaf_soy_R4_T2_filt, "ANCOM/Input/fungi_leaf_soy_R4_T2_filt_022621.rds")
saveRDS(fungi_leaf_soy_R6_T2_filt, "ANCOM/Input/fungi_leaf_soy_R6_T2_filt_022621.rds")
```


```{r}
#### CORN V8 ####
fungi_leaf_corn_V8_T1_ANCOM <- readRDS("ANCOM/HPCC_output/fungi_leaf_corn_V8_T1_ANCOM.rds")
colnames(fungi_leaf_corn_V8_T1_ANCOM) <- c("OTU", "W", "detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")
fungi_leaf_corn_V8_T1_ANCOM$GrowthStage <- "V8"
fungi_leaf_corn_V8_T1_ANCOM$Crop <- "Corn"
fungi_leaf_corn_V8_T1_ANCOM$Treatment <- "T1"
fungi_leaf_corn_V8_T1_ANCOM$StructuralZero <- ifelse(fungi_leaf_corn_V8_T1_ANCOM$W == Inf, TRUE, FALSE)
fungi_leaf_corn_V8_T1_ANCOM$OTU[fungi_leaf_corn_V8_T1_ANCOM$detected_0.7 == TRUE & fungi_leaf_corn_V8_T1_ANCOM$StructuralZero == FALSE]
fungi_leaf_corn_V8_T1_sig <- fungi_leaf_corn_V8_T1_ANCOM$OTU[fungi_leaf_corn_V8_T1_ANCOM$detected_0.7 == TRUE]

fungi_leaf_corn_V8_T2_ANCOM <- readRDS("ANCOM/HPCC_output/fungi_leaf_corn_V8_T2_ANCOM.rds")
colnames(fungi_leaf_corn_V8_T2_ANCOM) <- c("OTU", "W", "detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")
fungi_leaf_corn_V8_T2_ANCOM$GrowthStage <- "V8"
fungi_leaf_corn_V8_T2_ANCOM$Crop <- "Corn"
fungi_leaf_corn_V8_T2_ANCOM$Treatment <- "T2"
fungi_leaf_corn_V8_T2_ANCOM$StructuralZero <- ifelse(fungi_leaf_corn_V8_T2_ANCOM$W == Inf, TRUE, FALSE)
fungi_leaf_corn_V8_T2_ANCOM$OTU[fungi_leaf_corn_V8_T2_ANCOM$detected_0.7 == TRUE & fungi_leaf_corn_V8_T2_ANCOM$StructuralZero == FALSE]
fungi_leaf_corn_V8_T2_sig <- fungi_leaf_corn_V8_T2_ANCOM$OTU[fungi_leaf_corn_V8_T2_ANCOM$detected_0.7 == TRUE]

#### SOYBEAN BOTH GROWTH STAGES ####
fungi_leaf_soy_R4_T1_ANCOM <- readRDS("ANCOM/HPCC_output/fungi_leaf_soy_R4_T1_ANCOM.rds")
colnames(fungi_leaf_soy_R4_T1_ANCOM) <- c("OTU", "W", "detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")
fungi_leaf_soy_R4_T1_ANCOM$GrowthStage <- "R4"
fungi_leaf_soy_R4_T1_ANCOM$Crop <- "Soy"
fungi_leaf_soy_R4_T1_ANCOM$Treatment <- "T1"
fungi_leaf_soy_R4_T1_ANCOM$StructuralZero <- ifelse(fungi_leaf_soy_R4_T1_ANCOM$W == Inf, TRUE, FALSE)
fungi_leaf_soy_R4_T1_ANCOM$OTU[fungi_leaf_soy_R4_T1_ANCOM$detected_0.7 == TRUE & fungi_leaf_soy_R4_T1_ANCOM$StructuralZero == FALSE]
fungi_leaf_soy_R4_T1_sig <- fungi_leaf_soy_R4_T1_ANCOM$OTU[fungi_leaf_soy_R4_T1_ANCOM$detected_0.7 == TRUE]

fungi_leaf_soy_R4_T2_ANCOM <- readRDS("ANCOM/HPCC_output/fungi_leaf_soy_R4_T2_ANCOM.rds")
colnames(fungi_leaf_soy_R4_T2_ANCOM) <- c("OTU", "W", "detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")
fungi_leaf_soy_R4_T2_ANCOM$GrowthStage <- "R4"
fungi_leaf_soy_R4_T2_ANCOM$Crop <- "Soy"
fungi_leaf_soy_R4_T2_ANCOM$Treatment <- "T2"
fungi_leaf_soy_R4_T2_ANCOM$StructuralZero <- ifelse(fungi_leaf_soy_R4_T2_ANCOM$W == Inf, TRUE, FALSE)
fungi_leaf_soy_R4_T2_ANCOM$OTU[fungi_leaf_soy_R4_T2_ANCOM$detected_0.7 == TRUE & fungi_leaf_soy_R4_T2_ANCOM$StructuralZero == FALSE]
fungi_leaf_soy_R4_T2_sig <- fungi_leaf_soy_R4_T2_ANCOM$OTU[fungi_leaf_soy_R4_T2_ANCOM$detected_0.7 == TRUE]

fungi_leaf_soy_R6_T1_ANCOM <- readRDS("ANCOM/HPCC_output/fungi_leaf_soy_R6_T1_ANCOM.rds")
colnames(fungi_leaf_soy_R6_T1_ANCOM) <- c("OTU", "W", "detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")
fungi_leaf_soy_R6_T1_ANCOM$GrowthStage <- "R6"
fungi_leaf_soy_R6_T1_ANCOM$Crop <- "Soy"
fungi_leaf_soy_R6_T1_ANCOM$Treatment <- "T1"
fungi_leaf_soy_R6_T1_ANCOM$StructuralZero <- ifelse(fungi_leaf_soy_R6_T1_ANCOM$W == Inf, TRUE, FALSE)
fungi_leaf_soy_R6_T1_ANCOM$OTU[fungi_leaf_soy_R6_T1_ANCOM$detected_0.7 == TRUE & fungi_leaf_soy_R6_T1_ANCOM$StructuralZero == FALSE]
fungi_leaf_soy_R6_T1_sig <- fungi_leaf_soy_R6_T1_ANCOM$OTU[fungi_leaf_soy_R6_T1_ANCOM$detected_0.7 == TRUE]

fungi_leaf_soy_R6_T2_ANCOM <- readRDS("ANCOM/HPCC_output/fungi_leaf_soy_R6_T2_ANCOM.rds")
colnames(fungi_leaf_soy_R6_T2_ANCOM) <- c("OTU", "W", "detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")
fungi_leaf_soy_R6_T2_ANCOM$GrowthStage <- "R6"
fungi_leaf_soy_R6_T2_ANCOM$Crop <- "Soy"
fungi_leaf_soy_R6_T2_ANCOM$Treatment <- "T2"
fungi_leaf_soy_R6_T2_ANCOM$StructuralZero <- ifelse(fungi_leaf_soy_R6_T2_ANCOM$W == Inf, TRUE, FALSE)
fungi_leaf_soy_R6_T2_ANCOM$OTU[fungi_leaf_soy_R6_T2_ANCOM$detected_0.7 == TRUE & fungi_leaf_soy_R6_T2_ANCOM$StructuralZero == FALSE]
fungi_leaf_soy_R6_T2_sig <- fungi_leaf_soy_R6_T2_ANCOM$OTU[fungi_leaf_soy_R6_T2_ANCOM$detected_0.7 == TRUE]
```

### Composition of fungicide affected OTUs
```{r}
#### DIFF % RA SOY ####
fungi.ra.soy <- fungi.no.norm.filt %>%  
  subset_samples(Compartment == "Leaf" & Crop  == "Soy") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt.fast() %>%                                         # Melt to long format
  group_by(Fungicide, DateSampled, GrowthStage, Crop, Treatment) %>%
  arrange(OTU)    # Sort data frame alphabetically by OTU
mean.rel.abund <- fungi.ra.soy %>% 
  group_by(OTU, Fungicide, DateSampled, GrowthStage, Crop, Treatment) %>% 
  nest() %>%
  mutate(mean.relabund = purrr::map(data,~mean(.$Abundance))) %>%
  mutate(median.relabund = purrr::map(data,~median(.$Abundance))) %>%
  mutate(SE.relabund = purrr::map(data,~sd(.$Abundance)/sqrt(length(.$Abundance)))) %>%
  unnest(c(mean.relabund, SE.relabund, median.relabund))

mean.rel.abund.fungicide.over.time.soy <- pivot_wider(mean.rel.abund, id_cols = c(OTU,Fungicide,GrowthStage, DateSampled, Crop,Treatment), names_from = Fungicide, values_from = c(mean.relabund)) # put the F and C abundances side-by-side
mean.rel.abund.fungicide.over.time.soy$DateSampled <- as.Date(mean.rel.abund.fungicide.over.time.soy$DateSampled, "%d-%b-%y") # Change date to correct format
mean.rel.abund.fungicide.over.time.soy$diff <- mean.rel.abund.fungicide.over.time.soy$F - mean.rel.abund.fungicide.over.time.soy$C


#### DIFF % RA CORN ####
fungi.ra.corn <- fungi.no.norm.filt %>%  
  subset_samples(Compartment == "Leaf" & Crop  == "Corn") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  phyloseq::transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt.fast() %>%                                         # Melt to long format
  group_by(Fungicide, DateSampled, GrowthStage, Crop, Treatment) %>%
  arrange(OTU)    # Sort data frame alphabetically by OTU
mean.rel.abund <- fungi.ra.corn %>% 
  group_by(OTU, Fungicide, DateSampled, GrowthStage, Crop, Treatment) %>% 
  nest() %>%
  mutate(mean.relabund = purrr::map(data,~mean(.$Abundance))) %>%
  mutate(median.relabund = purrr::map(data,~median(.$Abundance))) %>%
  mutate(SE.relabund = purrr::map(data,~sd(.$Abundance)/sqrt(length(.$Abundance)))) %>%
  unnest(c(mean.relabund, SE.relabund, median.relabund))

mean.rel.abund.fungicide.over.time.corn <- pivot_wider(mean.rel.abund, id_cols = c(OTU,Fungicide,GrowthStage, DateSampled, Crop,Treatment), names_from = Fungicide, values_from = c(mean.relabund)) # put the F and C abundances side-by-side
mean.rel.abund.fungicide.over.time.corn$DateSampled <- as.Date(mean.rel.abund.fungicide.over.time.corn$DateSampled, "%d-%b-%y") # Change date to correct format
mean.rel.abund.fungicide.over.time.corn$diff <- mean.rel.abund.fungicide.over.time.corn$F - mean.rel.abund.fungicide.over.time.corn$C


```

Here we are filtering out the OTUs that were called deferentially abundant, but were never present in either the control or fungicide treated plots. 
```{r}
diff.abundant.fotus.leaf.soy.t1 <- unique(c(as.character(fungi_leaf_soy_R4_T1_sig), as.character(fungi_leaf_soy_R6_T1_sig)))

t1.diff.abund.soy.filter <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T1" & OTU %in% diff.abundant.fotus.leaf.soy.t1) %>%
  left_join(taxonomy, by = c("OTU" = "OTU_ID")) %>%
  group_by(OTU) %>%
  nest() %>%
  mutate(sum.fungicide = purrr::map(data,~sum(.$F))) %>%
  mutate(sum.control = purrr::map(data,~sum(.$C))) %>%
  unnest(c(sum.fungicide, sum.control)) %>%
  subset(sum.fungicide == 0 | sum.control == 0) %>%
  select(OTU) %>%
  as.matrix() %>%
  as.character()

diff.abundant.fotus.leaf.soy.t2 <- unique(c(as.character(fungi_leaf_soy_R4_T2_sig), as.character(fungi_leaf_soy_R6_T2_sig)))

t2.diff.abund.soy.filter <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T2" & OTU %in% diff.abundant.fotus.leaf.soy.t2) %>%
  left_join(taxonomy, by = c("OTU" = "OTU_ID")) %>%
  group_by(OTU) %>%
  nest() %>%
  mutate(sum.fungicide = purrr::map(data,~sum(.$F))) %>%
  mutate(sum.control = purrr::map(data,~sum(.$C))) %>%
  unnest(c(sum.fungicide, sum.control)) %>%
  subset(sum.fungicide == 0 | sum.control == 0) %>%
  select(OTU) %>%
  as.matrix() %>%
  as.character()

diff.abundant.fotus.leaf.corn.t1 <- unique(c(as.character(fungi_leaf_corn_V8_T1_sig)))

t1.diff.abund.corn.filter <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T1" & OTU %in% diff.abundant.fotus.leaf.corn.t1) %>%
  left_join(taxonomy, by = c("OTU" = "OTU_ID")) %>%
  group_by(OTU) %>%
  nest() %>%
  mutate(sum.fungicide = purrr::map(data,~sum(.$F))) %>%
  mutate(sum.control = purrr::map(data,~sum(.$C))) %>%
  unnest(c(sum.fungicide, sum.control)) %>%
  subset(sum.fungicide == 0 | sum.control == 0) %>%
  select(OTU) %>%
  as.matrix() %>%
  as.character()

diff.abundant.fotus.leaf.corn.t2 <- unique(c(as.character(fungi_leaf_corn_V8_T2_sig)))

t2.diff.abund.corn.filter <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T2" & OTU %in% diff.abundant.fotus.leaf.corn.t2) %>%
  left_join(taxonomy, by = c("OTU" = "OTU_ID")) %>%
  group_by(OTU) %>%
  nest() %>%
  mutate(sum.fungicide = purrr::map(data,~sum(.$F))) %>%
  mutate(sum.control = purrr::map(data,~sum(.$C))) %>%
  unnest(c(sum.fungicide, sum.control)) %>%
  subset(sum.fungicide == 0 | sum.control == 0) %>%
  select(OTU) %>%
  as.matrix() %>%
  as.character()

```


```{r}
#### COMPOSITION of Fungicide affected OTUS Soy #####

ANCOM.combined.soy.T1 <- rbind.data.frame(fungi_leaf_soy_R4_T1_ANCOM, fungi_leaf_soy_R6_T1_ANCOM) %>%
  subset(OTU %!in% t1.diff.abund.soy.filter) %>%
  left_join(taxonomy, by = "OTU") %>%
  subset(detected_0.7 == "TRUE")

ANCOM.combined.soy.T2 <- rbind.data.frame(fungi_leaf_soy_R4_T2_ANCOM, fungi_leaf_soy_R6_T2_ANCOM) %>%
  subset(OTU %!in% t2.diff.abund.soy.filter) %>%
  left_join(taxonomy, by = "OTU") %>%
  subset(detected_0.7 == "TRUE")

ANCOM.combined2.filt.soy <- rbind.data.frame(ANCOM.combined.soy.T1, ANCOM.combined.soy.T2)

mean.rel.abund.fungicide.over.time2.soy <- left_join(mean.rel.abund.fungicide.over.time.soy, ANCOM.combined2.filt.soy, by = c("OTU", "Treatment", "Crop", "GrowthStage")) %>%
  na.omit()

mean.rel.abund.fungicide.over.time2.soy$Direction <- ifelse(mean.rel.abund.fungicide.over.time2.soy$diff < 0, "Decrease", "Increase")

soy.T1.fungicide.decrease <- mean.rel.abund.fungicide.over.time2.soy$OTU[mean.rel.abund.fungicide.over.time2.soy$Direction == "Decrease" & mean.rel.abund.fungicide.over.time2.soy$Crop == "Soy" & mean.rel.abund.fungicide.over.time2.soy$Treatment == "T1"]
soy.T1.fungicide.increase <- mean.rel.abund.fungicide.over.time2.soy$OTU[mean.rel.abund.fungicide.over.time2.soy$Direction == "Increase" & mean.rel.abund.fungicide.over.time2.soy$Crop == "Soy" & mean.rel.abund.fungicide.over.time2.soy$Treatment == "T1"]

mean.rel.abund.fungicide.over.time3.soy <- mean.rel.abund.fungicide.over.time2.soy %>%
  group_by(GrowthStage, Crop, Direction, Class, Treatment) %>%
  nest() %>%
  mutate(sum.fungicide.otus = purrr::map(data,~nrow(.))) %>%
  unnest(sum.fungicide.otus)

mean.rel.abund.fungicide.over.time3.soy$bar.count <- ifelse(mean.rel.abund.fungicide.over.time3.soy$Direction == "Decrease", -mean.rel.abund.fungicide.over.time3.soy$sum.fungicide.otus, mean.rel.abund.fungicide.over.time3.soy$sum.fungicide.otus)

class.label <- c("Agaricomycetes", "Agaricostilbomycetes", "Dothideomycetes", "Tremellomycetes", "Sordariomycetes", "Eurotiomycetes", "Exobasidiomycetes", "Microbotryomycetes", "unidentified")

mean.rel.abund.fungicide.over.time3.soy$class.label <- ifelse(mean.rel.abund.fungicide.over.time3.soy$Class %!in% class.label, "Other", mean.rel.abund.fungicide.over.time3.soy$Class)


#### COMPOSITION of Fungicide affected OTUS Corn #####

ANCOM.combined.corn.T1 <- rbind.data.frame(fungi_leaf_corn_V8_T1_ANCOM) %>%
  subset(OTU %!in% t1.diff.abund.soy.filter) %>%
  left_join(taxonomy, by = "OTU") %>%
  subset(detected_0.7 == "TRUE")

ANCOM.combined.corn.T2 <- rbind.data.frame(fungi_leaf_corn_V8_T2_ANCOM) %>%
  subset(OTU %!in% t2.diff.abund.soy.filter) %>%
  left_join(taxonomy, by = "OTU") %>%
  subset(detected_0.7 == "TRUE")

ANCOM.combined2.filt.corn <- rbind.data.frame(ANCOM.combined.corn.T1, ANCOM.combined.corn.T2)

mean.rel.abund.fungicide.over.time2.corn <- left_join(mean.rel.abund.fungicide.over.time.corn, ANCOM.combined2.filt.corn, by = c("OTU", "Treatment", "Crop", "GrowthStage")) %>%
  na.omit()

mean.rel.abund.fungicide.over.time2.corn$Direction <- ifelse(mean.rel.abund.fungicide.over.time2.corn$diff < 0, "Decrease", "Increase")

corn.T1.fungicide.decrease <- mean.rel.abund.fungicide.over.time2.corn$OTU[mean.rel.abund.fungicide.over.time2.corn$Direction == "Decrease" & mean.rel.abund.fungicide.over.time2.corn$Crop == "Corn" & mean.rel.abund.fungicide.over.time2.corn$Treatment == "T1"]
corn.T1.fungicide.increase <- mean.rel.abund.fungicide.over.time2.corn$OTU[mean.rel.abund.fungicide.over.time2.corn$Direction == "Increase" & mean.rel.abund.fungicide.over.time2.corn$Crop == "Corn" & mean.rel.abund.fungicide.over.time2.corn$Treatment == "T1"]

mean.rel.abund.fungicide.over.time3.corn <- mean.rel.abund.fungicide.over.time2.corn %>%
  group_by(GrowthStage, Crop, Direction, Class, Treatment) %>%
  nest() %>%
  mutate(sum.fungicide.otus = purrr::map(data,~nrow(.))) %>%
  unnest(sum.fungicide.otus)

mean.rel.abund.fungicide.over.time3.corn$bar.count <- ifelse(mean.rel.abund.fungicide.over.time3.corn$Direction == "Decrease", -mean.rel.abund.fungicide.over.time3.corn$sum.fungicide.otus, mean.rel.abund.fungicide.over.time3.corn$sum.fungicide.otus)

class.label <- c("Agaricomycetes", "Agaricostilbomycetes", "Dothideomycetes", "Tremellomycetes", "Sordariomycetes", "Eurotiomycetes", "Exobasidiomycetes", "Microbotryomycetes", "unidentified")

mean.rel.abund.fungicide.over.time3.corn$class.label <- ifelse(mean.rel.abund.fungicide.over.time3.corn$Class %!in% class.label, "Other", mean.rel.abund.fungicide.over.time3.corn$Class)
```


### What are the top classes of taxa that were differentially abundant
```{r}
diff.fotus <- data.frame(unique(c(ANCOM.combined2.filt.soy$OTU, ANCOM.combined2.filt.corn$OTU)))
duplicated(diff.fotus)
colnames(diff.fotus) <- "OTU_ID"
diff.fotus2 <- left_join(diff.fotus, taxonomy, by = "OTU_ID")

diff.fotus2 %>%
  group_by(Phylum) %>%
  tally(sort = T) %>%
  mutate(freq = n / sum(n)) %>%
  print(n=22)

diff.fotus2 %>%
  group_by(Class) %>%
  tally(sort = T) %>%
  mutate(freq = n / sum(n)) %>%
  print(n=22)

diff.fotus2 %>%
  subset(Class == "Tremellomycetes") %>%
  group_by(Family) %>%
  tally(sort = T) %>%
  mutate(freq = n / sum(n))

diff.fotus2 %>%
  subset(Class == "Dothideomycetes") %>%
  group_by(Family) %>%
  tally(sort = T) %>%
  mutate(freq = n / sum(n))

diff.fotus2 %>%
  subset(Class == "Sordariomycetes") %>%
  group_by(Genus) %>%
  tally(sort = T) %>%
  mutate(freq = n / sum(n))

# fold-change decrease in relative abundance
mean.rel.abund.fungicide.over.time2.soy %>%
  group_by(GrowthStage, Treatment) %>%
  subset(diff < 0) %>%
  mutate(fold.chage = F/C) %>%
  select(Taxonomy, fold.chage) %>%
  as.data.frame()

# fold-change increase in relative abundance
mean.rel.abund.fungicide.over.time2.soy %>%
  group_by(GrowthStage, Treatment) %>%
  subset(diff > 0) %>%
  mutate(fold.chage = F/C) %>%
  select(Taxonomy, fold.chage) %>%
  as.data.frame()

mean.rel.abund.fungicide.over.time2 <- rbind.data.frame(mean.rel.abund.fungicide.over.time2.soy, mean.rel.abund.fungicide.over.time2.corn)

mean.rel.abund.fungicide.over.time2 %>%
  ungroup() %>%
  subset(Class == "Tremellomycetes") %>%
  subset(diff < 0) %>%
  select(OTU) %>%
  unique() %>%
  left_join(taxonomy, by = "OTU") %>%
  group_by(Family) %>%
  tally(sort = T) %>%
  mutate(freq = n / sum(n))

#Supplemental Table S
write.csv(mean.rel.abund.fungicide.over.time2, "fungicide_diff_abund.csv")
```


### Plot the composition of differentially abundant OTUs
```{r}
mean.rel.abund.fungicide.over.time.combined <- rbind.data.frame(mean.rel.abund.fungicide.over.time3.soy, mean.rel.abund.fungicide.over.time3.corn)
mean.rel.abund.fungicide.over.time.combined$x_axis <- interaction(mean.rel.abund.fungicide.over.time.combined$Crop, mean.rel.abund.fungicide.over.time.combined$GrowthStage, mean.rel.abund.fungicide.over.time.combined$Treatment)

mean.rel.abund.fungicide.over.time.combined$x_axis <- factor(mean.rel.abund.fungicide.over.time.combined$x_axis,levels = c("Soy.R4.T1", "Soy.R6.T1", "Soy.R4.T2", "Soy.R6.T2", "Corn.V8.T1", "Corn.V8.T2"))



#### PLOT COMPOSITION OF DIFFERENTIALLY ABUNDANT OTUS #### 
Fig.2A.Composition.ANCOM <- ggplot(mean.rel.abund.fungicide.over.time.combined, aes(x = x_axis, y = bar.count)) +
  geom_bar(aes(fill = class.label), stat = "identity") +
  theme_classic() +
  xlab("") + 
  ylab("OTU count") + 
  ggtitle("Fungi") +
  scale_fill_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[4], cbbPalette[5], cbbPalette[8], "grey", cbbPalette[6], cbbPalette[7], "purple")) + 
  scale_x_discrete(labels = c("Soy Conv. 13-dpf", "Soy Conv. 33-dpf", "Soy No-till 13-dpf", "Soy No-till 33-dpf", "Corn Conv. 9-dpf", "Corn No-till 9-dpf")) +
  geom_hline(yintercept = 0, lty = 2) +
  ylim(c(-80, 15)) +
theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="bottom",
   legend.title = element_blank(), 
   axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 
  #annotate("text", x = 3, y = 15, label = "Increased abundance") +
  #annotate("text", x = 3, y = -80, label = "Decreased abundance")
```

```{r}
mean.rel.abund.fungicide.over.time2.combined <- rbind.data.frame(mean.rel.abund.fungicide.over.time2.soy, mean.rel.abund.fungicide.over.time2.corn)
mean.rel.abund.fungicide.over.time4.combined <- mean.rel.abund.fungicide.over.time2.combined %>%
  group_by(GrowthStage, Crop, Direction, Class, Family, Treatment) %>%
  nest() %>%
  mutate(sum.fungicide.otus = purrr::map(data,~nrow(.))) %>%
  unnest(sum.fungicide.otus)

mean.rel.abund.fungicide.over.time4.combined$bar.count <- ifelse(mean.rel.abund.fungicide.over.time4.combined$Direction == "Decrease", -mean.rel.abund.fungicide.over.time4.combined$sum.fungicide.otus, mean.rel.abund.fungicide.over.time4.combined$sum.fungicide.otus)

mean.rel.abund.fungicide.over.time4.combined$x_axis <- interaction(mean.rel.abund.fungicide.over.time4.combined$Crop, mean.rel.abund.fungicide.over.time4.combined$GrowthStage, mean.rel.abund.fungicide.over.time4.combined$Treatment)

mean.rel.abund.fungicide.over.time4.combined$x_axis <- factor(mean.rel.abund.fungicide.over.time4.combined$x_axis,levels = c("Soy.R4.T1", "Soy.R6.T1", "Soy.R4.T2", "Soy.R6.T2", "Corn.V8.T1", "Corn.V8.T2"))

Fig.2B.Soy.Trem.Composition.ANCOM <- mean.rel.abund.fungicide.over.time4.combined %>%
  subset(Class == "Tremellomycetes") %>%
ggplot(aes(x = x_axis, y = bar.count)) +
  geom_bar(aes(fill = Family), stat = "identity") +
  theme_classic() + 
  xlab("") + 
  ylab("OTU count") + 
  ggtitle("Tremellomycetes") +
  scale_fill_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[4], cbbPalette[5], cbbPalette[8], "grey", cbbPalette[6], cbbPalette[7], "purple")) + 
  scale_x_discrete(labels = c("Soy Conv. 13-dpf", "Soy Conv. 33-dpf", "Soy No-till 13-dpf", "Soy No-till 33-dpf", "Corn Conv. 9-dpf", "Corn No-till 9-dpf")) +
  geom_hline(yintercept = 0, lty = 2) +
  ylim(c(-30, 5)) +
theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.title = element_blank(),
   legend.position="bottom",
   axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) 
  #annotate("text", x = 3, y = 5, label = "Increased abundance") +
  #annotate("text", x = 3, y = -30, label = "Decreased abundance")
```

### Recovered or non-recovered OTUs 
- Corn communities recovered fully 
- Soybeans did not 
- This code only takes those that were significantly different, it does not tell which increased or decreased in abundance
```{r}
non_recovered_soy_T1 <- intersect(fungi_leaf_soy_R4_T1_sig, fungi_leaf_soy_R6_T1_sig) # the OTUs that were sig diff after R4 and R6
recovered_soy_T1 <- setdiff(fungi_leaf_soy_R4_T1_sig, fungi_leaf_soy_R6_T1_sig)# the OTUs that were significant after R4 but not on R6
secondary_nonrecovered_soy_T1 <- setdiff(fungi_leaf_soy_R6_T1_sig, fungi_leaf_soy_R4_T1_sig)# the OTUs that were significant after R6 but not on R4
sig.diff.T1 <- union(fungi_leaf_soy_R4_T1_sig, fungi_leaf_soy_R6_T1_sig) # the total list of OTUs that was significant at R4 or R6

# Sanity check - T1 soybean
all(sort(c(setdiff(fungi_leaf_soy_R4_T1_sig,fungi_leaf_soy_R6_T1_sig),
      intersect(fungi_leaf_soy_R6_T1_sig,fungi_leaf_soy_R4_T1_sig),
      setdiff(fungi_leaf_soy_R6_T1_sig,fungi_leaf_soy_R4_T1_sig))) ==
   sort(union(fungi_leaf_soy_R4_T1_sig,fungi_leaf_soy_R6_T1_sig)))


sort(c(soy.T1.fungicide.decrease, soy.T1.fungicide.increase))


non_recovered_soy_T2 <- intersect(fungi_leaf_soy_R4_T2_sig, fungi_leaf_soy_R6_T2_sig) # the OTUs that were sig diff after R4 and R6
recovered_soy_T2 <- setdiff(fungi_leaf_soy_R4_T2_sig, fungi_leaf_soy_R6_T2_sig)# the OTUs that were significant after R4 but not on R6
secondary_nonrecovered_soy_T2 <- setdiff(fungi_leaf_soy_R6_T2_sig, fungi_leaf_soy_R4_T2_sig)# the OTUs that were significant after R6 but not on R4
sig.diff.T2 <- union(fungi_leaf_soy_R4_T2_sig, fungi_leaf_soy_R6_T2_sig) # the total list of OTUs that was significant at R4 or R6

# Sanity check - T2 soybean
all(sort(c(setdiff(fungi_leaf_soy_R4_T2_sig,fungi_leaf_soy_R6_T2_sig),
      intersect(fungi_leaf_soy_R6_T2_sig,fungi_leaf_soy_R4_T2_sig),
      setdiff(fungi_leaf_soy_R6_T2_sig,fungi_leaf_soy_R4_T2_sig))) ==
   sort(union(fungi_leaf_soy_R4_T2_sig,fungi_leaf_soy_R6_T2_sig)))
```

### Plot the RA over-time of recovered and non-recovered OTUs
```{r}
#### CLR change ####
fungi.clr <- microbiome::transform(fungi.no.norm.filt, 'clr') %>%  
  subset_samples(Compartment == "Leaf") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  psmelt.fast() %>%                                         # Melt to long format
  group_by(Fungicide, Treatment, DateSampled, GrowthStage, Crop) %>%
  arrange(OTU)    # Sort data frame alphabetically by OTU
mean.rel.abund <- fungi.clr %>% 
  group_by(OTU, Fungicide, DateSampled, Treatment, GrowthStage, Crop) %>% 
  nest() %>%
  mutate(mean.relabund = purrr::map(data,~mean(.$Abundance))) %>%
  mutate(median.relabund = purrr::map(data,~median(.$Abundance))) %>%
  mutate(SE.relabund = purrr::map(data,~sd(.$Abundance)/sqrt(length(.$Abundance)))) %>%
  unnest(mean.relabund, SE.relabund, median.relabund)

mean.rel.abund.fungicide.over.time <- pivot_wider(mean.rel.abund, id_cols = c(OTU,Fungicide,GrowthStage, Treatment, DateSampled, Crop), names_from = Fungicide, values_from = c(mean.relabund)) # put the F and C abundances side-by-side
mean.rel.abund.fungicide.over.time$DateSampled <- as.Date(mean.rel.abund.fungicide.over.time$DateSampled, "%d-%b-%y") # Change date to correct format

mean.rel.abund.fungicide.over.time$diff <- mean.rel.abund.fungicide.over.time$F - mean.rel.abund.fungicide.over.time$C

class.label <- c("Agaricomycetes", "Agaricostilbomycetes", "Dothideomycetes", "Tremellomycetes", "Sordariomycetes", "Eurotiomycetes", "Exobasidiomycetes", "Microbotryomycetes", "unidentified")

## T1 soybean 
mean.rel.abund.fungicide.over.time %>%
  left_join(taxonomy, by = "OTU") %>%
  mutate(class.label = ifelse(Class %!in% class.label, "Other", Class)) %>%
  mutate(Recover = ifelse(OTU %in% non_recovered_soy_T1, "Non-recovered", 
                          ifelse(OTU %in% secondary_nonrecovered_soy_T1, "Indirect", "Recovered"))) %>%
  subset(Treatment == "T1" & Crop == "Soy" & OTU %in% c(non_recovered_soy_T1, recovered_soy_T1, secondary_nonrecovered_soy_T1)) %>%
ggplot() +
  geom_line(aes(x = DateSampled, y = diff, group = interaction(Recover, OTU), color = class.label, alpha = 0.6), size = 0.5) + 
  theme_bw() +
  facet_wrap(~Recover, ncol = 1) + 
  scale_color_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[4], cbbPalette[5], cbbPalette[8], "grey", cbbPalette[6], cbbPalette[7], "purple")) +  
  ylab("Mean CLR Difference") +
  theme(
  strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="bottom") + 
  geom_vline(xintercept=as.numeric(mean.rel.abund.fungicide.over.time.soy$DateSampled[c(1, 1, 284, 284, 2, 2, 283, 283, 3,3,285,285)]),
                linetype=4, colour="black") +
  geom_hline(yintercept= 0,
                linetype=4, colour="black")

## T2 soybean 
mean.rel.abund.fungicide.over.time %>%
  left_join(taxonomy, by = "OTU") %>%
  mutate(class.label = ifelse(Class %!in% class.label, "Other", Class)) %>%
  mutate(Recover = ifelse(OTU %in% non_recovered_soy_T2, "Non-recovered", 
                          ifelse(OTU %in% secondary_nonrecovered_soy_T2, "Indirect", "Recovered"))) %>%
  subset(Treatment == "T2" & Crop == "Soy" & OTU %in% c(non_recovered_soy_T2, recovered_soy_T2, secondary_nonrecovered_soy_T2)) %>%
ggplot() +
  geom_line(aes(x = DateSampled, y = diff, group = interaction(Recover, OTU), color = class.label, alpha = 0.6), size = 0.5) + 
  theme_bw() +
  facet_wrap(~Recover, ncol = 1) + 
  scale_color_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[4], cbbPalette[5], cbbPalette[8], "grey", cbbPalette[6], cbbPalette[7], "purple")) +  
  ylab("Mean CLR Difference") +
  theme(
  strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="bottom") + 
  geom_vline(xintercept=as.numeric(mean.rel.abund.fungicide.over.time.soy$DateSampled[c(1, 1, 284, 284, 2, 2, 283, 283, 3,3,285,285)]),
                linetype=4, colour="black") +
  geom_hline(yintercept= 0,
                linetype=4, colour="black")
```

Composition of recovered OTUs
```{r}
soy.no.till.classification <- as.data.frame(c(recovered_soy_T2, non_recovered_soy_T2, secondary_nonrecovered_soy_T2)) %>%
  mutate(OTU = c(recovered_soy_T2, non_recovered_soy_T2, secondary_nonrecovered_soy_T2)) %>%
  left_join(taxonomy, by = "OTU") %>% 
  mutate(class.label = ifelse(Class %!in% class.label, "Other", Class)) %>%
  mutate(Recover = ifelse(OTU %in% c(non_recovered_soy_T2, secondary_nonrecovered_soy_T2), "Non-recovered or Indirect", "Recovered")) %>%
  mutate(Treatment = "No-till")

soy.conv.classification <- as.data.frame(c(recovered_soy_T1, non_recovered_soy_T1, secondary_nonrecovered_soy_T1)) %>%
  mutate(OTU = c(recovered_soy_T1, non_recovered_soy_T1, secondary_nonrecovered_soy_T1)) %>%
  left_join(taxonomy, by = "OTU") %>% 
  mutate(class.label = ifelse(Class %!in% class.label, "Other", Class)) %>%
  mutate(Recover = ifelse(OTU %in% c(non_recovered_soy_T1, secondary_nonrecovered_soy_T1), "Non-recovered or Indirect", "Recovered")) %>%

  mutate(Treatment = "Conv.")

soy.classification <- rbind.data.frame(soy.no.till.classification[,-1], soy.conv.classification[,-1])
  
```

- I am defining the local extinctions as those OTUs in T1 or T2 whose abundance went to 0 following fungicide application (at R4) and was still zero at (R6) and was detected as deferentially abundant with ANCOM.
```{r}
Soy.T1.diff <- ANCOM.combined2.filt.soy %>%
  subset(Treatment == "T1") %>%
  select(OTU) %>%
  unique() %>%
  as.matrix() %>%
  as.character()

Soy.T2.diff <- ANCOM.combined2.filt.soy %>%
  subset(Treatment == "T2") %>%
  select(OTU) %>%
  unique() %>%
  as.matrix() %>%
  as.character()

Corn.T1.diff <- ANCOM.combined2.filt.corn %>%
  subset(Treatment == "T1") %>%
  select(OTU) %>%
  unique() %>%
  as.matrix() %>%
  as.character()

Corn.T2.diff <- ANCOM.combined2.filt.corn %>%
  subset(Treatment == "T2") %>%
  select(OTU) %>%
  unique() %>%
  as.matrix() %>%
  as.character()

# the corn needs to be done for both treatments
### Corn-T1 ####
present.V6.T1 <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T1" & GrowthStage == "V6" & F > 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.V8.T1 <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T1" & GrowthStage == "V8" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.V15.T1 <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T1" & GrowthStage == "V15" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

# Total Corn OTUs sig-different in either R6 and R4
local.extinction.putative <- intersect(present.V6.T1$OTU, intersect(absent.V8.T1$OTU, absent.V15.T1$OTU))

locally.extinct.fungicide.corn.T1 <- intersect(local.extinction.putative, Corn.T1.diff)

taxonomy[taxonomy$OTU %in% locally.extinct.fungicide.corn.T1,]

### Corn-T2 ####
present.V6.T2 <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T2" & GrowthStage == "V6" & F > 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.V8.T2 <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T2" & GrowthStage == "V8" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.V15.T2 <- mean.rel.abund.fungicide.over.time.corn %>%
  subset(Treatment == "T2" & GrowthStage == "V15" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

# Total Corn OTUs sig-different in either R6 and R4
local.extinction.putative <- intersect(present.V6.T2$OTU, intersect(absent.V8.T2$OTU, absent.V15.T2$OTU))

locally.extinct.fungicide.corn.T1 <- intersect(local.extinction.putative, Corn.T1.diff)

taxonomy[taxonomy$OTU %in% locally.extinct.fungicide.corn.T1,]

### Soybean-T1 ####
present.R3.T1 <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T1" & GrowthStage == "R3" & F > 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.R4.T1 <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T1" & GrowthStage == "R4" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.R6.T1 <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T1" & GrowthStage == "R6" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

# Total Soybean OTUs sig-different in either R6 and R4
local.extinction.putative <- intersect(present.R3.T1$OTU, intersect(absent.R4.T1$OTU, absent.R6.T1$OTU))

locally.extinct.fungicide.soy.T1 <- intersect(local.extinction.putative, Soy.T1.diff)

taxonomy[taxonomy$OTU %in% locally.extinct.fungicide.soy.T1,]

soy.conv.classification$local.exinction <- ifelse(soy.conv.classification$OTU %in% locally.extinct.fungicide.soy.T1, "Local Extinction", soy.conv.classification$Recover)


### Soybean-T2 ####
present.R3.T2 <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T2" & GrowthStage == "R3" & F > 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.R4.T2 <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T2" & GrowthStage == "R4" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

absent.R6.T2 <- mean.rel.abund.fungicide.over.time.soy %>%
  subset(Treatment == "T2" & GrowthStage == "R6" & F == 0 & C > 0) %>%
  as.data.frame() %>%
  select(OTU) 

# Total Soybean OTUs sig-different in either R6 and R4
local.extinction.putative <- intersect(present.R3.T2$OTU, intersect(absent.R4.T2$OTU, absent.R6.T2$OTU))

locally.extinct.fungicide.soy.T2 <- intersect(local.extinction.putative, Soy.T1.diff)

taxonomy[taxonomy$OTU %in% locally.extinct.fungicide.soy.T2,]

soy.no.till.classification$local.exinction <- ifelse(soy.no.till.classification$OTU %in% locally.extinct.fungicide.soy.T2, "Local Extinction", soy.no.till.classification$Recover)

soy.classification <- rbind.data.frame(soy.no.till.classification[,-1], soy.conv.classification[,-1])
```

A good majority of the Tremmellomycetes were within the Bulleribacideaceae family, lets plot that. 
```{r}
saveRDS(soy.classification, "soy_classification.RDS")


Fig.2C.Soy.Recover.ANCOM <- ggplot(soy.classification, aes(x = interaction(local.exinction, Treatment), fill = class.label)) +
  geom_bar() +
  theme_classic() + 
  xlab("") + 
  ylab("OTU count") + 
  ggtitle("Fungi - Soybean") +
  scale_fill_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[4], cbbPalette[5], cbbPalette[8], "grey", cbbPalette[6], cbbPalette[7], "purple")) + 
  scale_x_discrete(labels = c("Local Extinction", "Non-recovered \n or Indirect", "Recovered", 
                              "Local Extinction", "Non-recovered \n or Indirect", "Recovered")) +
theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="none",
   legend.title = element_blank(),
   axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  facet_wrap(~Treatment, scales = "free_x") 

Trem.OTUs.recov <- soy.classification %>%
  subset(Class == "Tremellomycetes") %>%
ggplot(aes(x = interaction(local.exinction, Treatment), fill = Family)) +
  geom_bar() +
  theme_classic() + 
  xlab("") + 
  ylab("OTU count") + 
  ggtitle("Tremellomycetes - Soybean") +
scale_fill_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[5], cbbPalette[8], cbbPalette[6], cbbPalette[7], "purple")) + 
  scale_x_discrete(drop = FALSE, labels = c("Local Extinction", "Non-recovered \n or Indirect", "Recovered", 
                                            "Local Extinction", "Non-recovered \n or Indirect", "Recovered")) +
theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   #legend.position="none",
   legend.title = element_blank(),
   axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  annotate("text", x = 2, y = 25, label = "Conv.") +
  annotate("text", x = 5, y = 25, label = "No-till") 
  #stat_count(geom = "text", aes(label = stat(count)), colour="black")
```

Lets plot the bulleribacideaceae richness. 
```{r}
physeq_fungi_no_norm_bull <- fungi.no.norm.filt %>% 
  subset_taxa(Family=="Bulleribasidiaceae") %>%
  subset_samples(Compartment == "Leaf") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

physeq_fungi_no_norm_bull@sam_data$richness <- estimate_richness(physeq_fungi_no_norm_bull, measures=c("Observed"))$Observed
physeq_fungi_no_norm_bull_samples <- data.frame(physeq_fungi_no_norm_bull@sam_data)
physeq_fungi_no_norm_bull_samples$DateSampled <- as.Date(physeq_fungi_no_norm_bull_samples$DateSampled, "%d-%b-%y")

physeq_fungi_no_norm_bull_samples$Treatment <- ifelse(physeq_fungi_no_norm_bull_samples$Treatment == "T1", "Conv.", "No-till")

saveRDS(physeq_fungi_no_norm_bull_samples, "Bull_richness")

soybean.bull.richness <- ggplot(physeq_fungi_no_norm_bull_samples[physeq_fungi_no_norm_bull_samples$Crop == "Soy",], aes(x = DateSampled, y = richness, group = interaction(Fungicide, Treatment), color = Fungicide)) + 
  geom_jitter(width = 0.5, alpha = 0.5) +
  stat_summary(fun.y=mean,geom="line") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  
  ylab("Bulleribasidiaceae \n richness") + 
  #ggtitle("Fungi") + 
  xlab("")+
  scale_color_manual(values=cbbPalette, label = c("Control", "Fungicide"), name = "") +
  #scale_linetype_manual(values = c(rep("solid", 2), rep("dashed", 2))) +
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = TRUE, size = 6, label.y = 30) +
  theme_classic() +
  facet_wrap(~Treatment, scales = "free_x") +
  theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="right",
   legend.title = element_blank()) + 
  ggtitle("Soybean")  
  

corn.bull.richness <- ggplot(physeq_fungi_no_norm_bull_samples[physeq_fungi_no_norm_bull_samples$Crop == "Corn",], aes(x = DateSampled, y = richness, group = interaction(Fungicide, Treatment), fill = Fungicide, color = Fungicide)) + 
  #geom_jitter(width = 0.5, shape = 21) +
  stat_summary(fun.y=mean,geom="line") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Bulleribasidiaceae \n richness") + 
  #ggtitle("Fungi") + 
  xlab("")+
  scale_color_manual(values=cbbPalette, label = c("Control", "Fungicide"), name = "") +
  scale_fill_manual(values=cbbPalette, label = c("Control", "Fungicide"), name = "") +
  #scale_linetype_manual(values = c(rep("solid", 2), rep("dashed", 2))) +
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = TRUE, size = 6, label.y = 35) +
  theme_classic() +
  ylim(c(0,40))+
  facet_wrap(~Treatment, scales = "free_x") +
  theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="right",
   legend.title = element_blank()) + 
  ggtitle("Corn")

```

##### CORE

### Priorizing the CORE microbiome as 
- those OTUs that contribute to 98% of the Bray-Curtis distances. 
- This code rely on rarefied reads. The function will rarefy. 
```{r}
core.prioritizing <- function(phyloseq.object){
  
set.seed(19)
rare.phyloseq.object <- rarefy_even_depth(phyloseq.object, replace=TRUE)

nReads=sample_sums(rare.phyloseq.object)[[1]]                                                                 # input dataset needs to be rarified and the rarifaction depth included 
otu <- rare.phyloseq.object@otu_table %>%
  as("matrix")
map <- rare.phyloseq.object@sam_data %>%
  as("data.frame")

otu_PA <- 1*((otu>0)==1)                                               # presence-absence data
otu_occ <- rowSums(otu_PA)/ncol(otu_PA)                                # occupancy calculation
otu_rel <- apply(decostand(otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
occ_abun <- add_rownames(as.data.frame(cbind(otu_occ, otu_rel)),'otu') # combining occupancy and abundance data frame

# Ranking OTUs based on their occupancy
# For caluclating raking index we included following conditions:
#   - time-specific occupancy (sumF) = frequency of detection within time point (genotype or site)
#   - replication consistency (sumG) = has occupancy of 1 in at least one time point (genotype or site) (1 if occupancy 1, else 0)

PresenceSum <- data.frame(otu = as.factor(row.names(otu)), otu) %>% 
  gather(SampleID, abun, -otu) %>%
  left_join(map, by = 'SampleID') %>%
  group_by(otu, GrowthStage) %>%
  dplyr::summarise(time_freq=sum(abun>0)/length(abun),            # frequency of detection between time points
            coreTime=ifelse(time_freq == 1, 1, 0)) %>%     # 1 only if occupancy 1 with specific time, 0 if not
  group_by(otu) %>%
  dplyr::summarise(sumF=sum(time_freq),
            sumG=sum(coreTime),
            nS=length(GrowthStage)*2,           
            Index=(sumF+sumG)/nS)                 # calculating weighting Index based on number of time points detected and 

otu_ranked <- occ_abun %>%
  left_join(PresenceSum, by='otu') %>%
  transmute(otu=otu,
            rank=Index) %>%
  arrange(desc(rank))

# Calculating the contribution of ranked OTUs to the BC similarity
BCaddition <- NULL

# calculating BC dissimilarity based on the 1st ranked OTU
# with 36 samples there should be 630 combinations n!/r!
otu_start=otu_ranked$otu[1]                   
start_matrix <- as.matrix(otu[otu_start,])
start_matrix <- t(start_matrix)
x <- apply(combn(ncol(start_matrix), 2), 2, function(x) sum(abs(start_matrix[,x[1]]- start_matrix[,x[2]]))/(2*nReads))
x_names <- apply(combn(ncol(start_matrix), 2), 2, function(x) paste(colnames(start_matrix)[x], collapse=' - '))
df_s <- data.frame(x_names,x)
df_s$rank_count <- 1
BCaddition <- rbind(BCaddition,df_s)
# calculating BC dissimilarity based on additon of ranked OTUs from 2nd to 500th. Can be set to the entire length of OTUs in the dataset, however it might take some time if more than 5000 OTUs are included.
for(i in 2:500){                              
  otu_add=otu_ranked$otu[i]                       
  add_matrix <- as.matrix(otu[otu_add,])
  add_matrix <- t(add_matrix)
  start_matrix <- rbind(start_matrix, add_matrix)
  x <- apply(combn(ncol(start_matrix), 2), 2, function(x) sum(abs(start_matrix[,x[1]]-start_matrix[,x[2]]))/(2*nReads))
  #x_names <- apply(combn(ncol(start_matrix), 2), 2, function(x) paste(colnames(start_matrix)[x], collapse=' - '))
  df_a <- data.frame(x_names,x)
  df_a$rank_count <- i 
  BCaddition <- rbind.data.frame(BCaddition, df_a)
}
# calculating the BC dissimilarity of the whole dataset (not needed if the second loop is already including all OTUs) 
x <-  apply(combn(ncol(otu), 2), 2, function(x) sum(abs(otu[,x[1]]-otu[,x[2]]))/(2*nReads))   
x_names <- apply(combn(ncol(otu), 2), 2, function(x) paste(colnames(otu)[x], collapse=' - '))
df_full <- data.frame(x_names,x)
df_full$rank_count <- length(rownames(otu))
BCfull <- rbind.data.frame(BCaddition, df_full)

BC_ranked <- BCfull %>%
  group_by(rank_count) %>%
  dplyr::summarise(MeanBC=mean(x)) %>%            # mean Bray-Curtis dissimilarity
  arrange(desc(-MeanBC)) %>%
  mutate(proportionBC=MeanBC/max(MeanBC))   # proportion of the dissimilarity explained by the n number of ranked OTUs
Increase=BC_ranked$MeanBC[-1]/BC_ranked$MeanBC[-length(BC_ranked$MeanBC)]
increaseDF <- data.frame(IncreaseBC=c(0,(Increase)), rank=factor(c(1:(length(Increase)+1))))
increaseDF$rank <- as.numeric(increaseDF$rank)
BC_ranked <- left_join(BC_ranked, increaseDF, by = c("rank_count" = "rank"))
BC_ranked <- BC_ranked[-nrow(BC_ranked),]

#Creating threshold for core inclusion - last call method

#B) Final increase in BC similarity of equal or greater then 2% 
lastCall <- last(as.numeric(BC_ranked$rank_count[(BC_ranked$IncreaseBC>=1.02)]))

#Creating plot of Bray-Curtis similarity
plot <- ggplot(BC_ranked[1:100,], aes(x=factor(BC_ranked$rank_count[1:100], levels=BC_ranked$rank_count[1:100]))) +
  geom_point(aes(y=proportionBC)) +
  theme_classic() + theme(strip.background = element_blank(),axis.text.x = element_text(size=7, angle=45)) +
  geom_vline(xintercept=last(as.numeric(BC_ranked$rank_count[(BC_ranked$IncreaseBC>=1.02)])), lty=3, col='black', cex=.5) +
  labs(x='ranked OTUs',y='Bray-Curtis similarity') +
  annotate(geom="text", x=last(as.numeric(BC_ranked$rank[(BC_ranked$IncreaseBC>=1.02)]))+3, y=.5, label=paste("Last 2% increase (",last(as.numeric(BC_ranked$rank[(BC_ranked$IncreaseBC>=1.02)])),")",sep=''), color="black")

core.otus.CSS.mean.T1 <- otu_ranked$otu[1:lastCall]
return_list <- list(core.otus.CSS.mean.T1, plot, otu_ranked, occ_abun)
return(return_list)
}

### Fungi ####
fungi.no.norm.leaf.soy.control.T1 <- fungi.no.norm.filt %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & Fungicide == "C" & Treatment == "T1") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.soy.T1 <- core.prioritizing(fungi.no.norm.leaf.soy.control.T1)

fungi.no.norm.leaf.soy.control.T2 <- fungi.no.norm.filt %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & Fungicide == "C" & Treatment == "T2") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.soy.T2 <- core.prioritizing(fungi.no.norm.leaf.soy.control.T2)

fungi.no.norm.leaf.corn.control.T1 <- fungi.no.norm.filt %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & Fungicide == "C" & Treatment == "T1") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.corn.T1 <- core.prioritizing(fungi.no.norm.leaf.corn.control.T1)

fungi.no.norm.leaf.corn.control.T2 <- fungi.no.norm.filt %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & Fungicide == "C" & Treatment == "T2") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.corn.T2 <- core.prioritizing(fungi.no.norm.leaf.corn.control.T2)

### Bacteria #### 
prok.no.norm.leaf.soy.control.T1 <- bacteria.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & Fungicide == "C" & Treatment == "T1") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.soy.T1.prok <- core.prioritizing(prok.no.norm.leaf.soy.control.T1)

prok.no.norm.leaf.soy.control.T2 <- bacteria.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & Fungicide == "C" & Treatment == "T2") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.soy.T2.prok <- core.prioritizing(prok.no.norm.leaf.soy.control.T2)

prok.no.norm.leaf.corn.control.T1 <- bacteria.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & Fungicide == "C" & Treatment == "T1") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.corn.T1.prok <- core.prioritizing(prok.no.norm.leaf.corn.control.T1)

prok.no.norm.leaf.corn.control.T2 <- bacteria.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & Fungicide == "C" & Treatment == "T2") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

core.rare.corn.T2.prok <- core.prioritizing(prok.no.norm.leaf.corn.control.T2)
```

```{r}
core.rare.soy.T1[[1]] 
core.rare.soy.T2[[1]]
core.rare.corn.T1[[1]]
core.rare.corn.T2[[1]]
```

The making of a supplemental table about the core memebers
```{r}
#### Fungi 
soy.t1.core.fungi <- data.frame(core.rare.soy.T1[[1]])
colnames(soy.t1.core.fungi) <- "OTU_ID"
soy.t1.core.fungi$Treatment <- "Conventional"
soy.t1.core.fungi$Crop <- "Soy"
soy.t1.core.fungi2 <- soy.t1.core.fungi %>%
  left_join(taxonomy, by = "OTU_ID")

soy.t2.core.fungi <- data.frame(core.rare.soy.T2[[1]])
colnames(soy.t2.core.fungi) <- "OTU_ID"
soy.t2.core.fungi$Treatment <- "No-till"
soy.t2.core.fungi$Crop <- "Soy"
soy.t2.core.fungi2 <- soy.t2.core.fungi %>%
  left_join(taxonomy, by = "OTU_ID")

corn.t1.core.fungi <- data.frame(core.rare.corn.T1[[1]])
colnames(corn.t1.core.fungi) <- "OTU_ID"
corn.t1.core.fungi$Treatment <- "Conventional"
corn.t1.core.fungi$Crop <- "Corn"
corn.t1.core.fungi2 <- corn.t1.core.fungi %>%
  left_join(taxonomy, by = "OTU_ID")

corn.t2.core.fungi <- data.frame(core.rare.corn.T2[[1]])
colnames(corn.t2.core.fungi) <- "OTU_ID"
corn.t2.core.fungi$Treatment <- "No-till"
corn.t2.core.fungi$Crop <- "Corn"
corn.t2.core.fungi2 <- corn.t2.core.fungi %>%
  left_join(taxonomy, by = "OTU_ID")

#### prokaryotes
soy.t1.core.prok <- data.frame(core.rare.soy.T1.prok[[1]])
colnames(soy.t1.core.prok) <- "OTU_ID"
soy.t1.core.prok$Treatment <- "Conventional"
soy.t1.core.prok$Crop <- "Soy"
soy.t1.core.prok2 <- soy.t1.core.prok %>%
  left_join(taxonomy.prok, by = "OTU_ID")

soy.t2.core.prok <- data.frame(core.rare.soy.T2.prok[[1]])
colnames(soy.t2.core.prok) <- "OTU_ID"
soy.t2.core.prok$Treatment <- "No-till"
soy.t2.core.prok$Crop <- "Soy"
soy.t2.core.prok2 <- soy.t2.core.prok %>%
  left_join(taxonomy.prok, by = "OTU_ID")

corn.t1.core.prok <- data.frame(core.rare.corn.T1.prok[[1]])
colnames(corn.t1.core.prok) <- "OTU_ID"
corn.t1.core.prok$Treatment <- "Conventional"
corn.t1.core.prok$Crop <- "Corn"
corn.t1.core.prok2 <- corn.t1.core.prok %>%
  left_join(taxonomy.prok, by = "OTU_ID")

corn.t2.core.prok <- data.frame(core.rare.corn.T2.prok[[1]])
colnames(corn.t2.core.prok) <- "OTU_ID"
corn.t2.core.prok$Treatment <- "No-till"
corn.t2.core.prok$Crop <- "Corn"
corn.t2.core.prok2 <- corn.t2.core.prok %>%
  left_join(taxonomy.prok, by = "OTU_ID")

prokaryote.cores <- rbind.data.frame(corn.t2.core.prok2, corn.t1.core.prok2, soy.t1.core.prok2, soy.t2.core.prok2)

fungi.cores <- rbind.data.frame(soy.t1.core.fungi2, soy.t2.core.fungi2, corn.t1.core.fungi2, corn.t2.core.fungi2)
```


```{r}
corn.no.till.classification.occ.abund <- left_join(core.rare.corn.T2[[4]], taxonomy, by = c("otu" = "OTU_ID"))
corn.no.till.classification.occ.abund$Core <- ifelse(corn.no.till.classification.occ.abund$OTU %in% core.rare.soy.T2[[1]], "Core", "Not-core")
corn.no.till.suptable <- corn.no.till.classification.occ.abund %>% 
  subset(otu %in% fungi_leaf_corn_V8_T2_sig)
corn.no.till.suptable$Treatment <- "No till"
corn.no.till.suptable$Recover <- "Recovered"

corn.conv.classification.occ.abund <- left_join(core.rare.corn.T1[[4]], taxonomy, by = c("otu" = "OTU_ID"))
corn.conv.classification.occ.abund$Core <- ifelse(corn.conv.classification.occ.abund$OTU %in% core.rare.soy.T1[[1]], "Core", "Not-core")
corn.conv.suptable <- corn.no.till.classification.occ.abund %>% 
  subset(otu %in% fungi_leaf_corn_V8_T1_sig)
corn.conv.suptable$Treatment <- "Conv."
corn.conv.suptable$Recover <- "Recovered"

corn.suptable <- rbind.data.frame(corn.conv.suptable, corn.no.till.suptable)

soy.no.till.classification.occ.abund <- left_join(soy.no.till.classification, core.rare.soy.T2[[4]], by = c("OTU" = "otu"))
soy.no.till.classification.occ.abund$Core <- ifelse(soy.no.till.classification.occ.abund$OTU %in% core.rare.soy.T2[[1]], "Core", "Not-core")

soy.conv.classification.occ.abund <- left_join(soy.conv.classification, core.rare.soy.T1[[4]], by = c("OTU" = "otu"))
soy.conv.classification.occ.abund$Core <- ifelse(soy.conv.classification.occ.abund$OTU %in% core.rare.soy.T1[[1]], "Core", "Not-core")

colnames(soy.no.till.classification.occ.abund)
colnames(soy.conv.classification.occ.abund)

soy.classification.combined <- rbind.data.frame(soy.no.till.classification.occ.abund[,-1], soy.conv.classification.occ.abund[,-1])

install.packages("ggbeeswarm")
library(ggbeeswarm)

supp.figure <- ggplot(soy.classification.combined, aes(x = interaction(local.exinction), y = otu_rel, fill = Core)) +
  geom_jitter(shape = 21)+
  #geom_beeswarm(shape = 21)+
  stat_compare_means(method = "kruskal") +
  #facet_wrap(~) + 
  xlab("") + 
  ylab("Occupancy") + 
  theme_classic()
```

```{r}
abund.occ.bull.soy <- soy.classification.combined %>% 
  subset(Family == "Bulleribasidiaceae") %>%
ggplot() + 
  geom_point(aes(x = log(otu_rel), y = otu_occ, fill = Genus, shape = interaction(local.exinction, Core)), size = 3, alpha = 0.8) + 
  #scale_color_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[5], cbbPalette[8], cbbPalette[6], cbbPalette[7], "purple"), name = "Bulleribasidiaceae \n Genera") +
    scale_fill_manual(values = c(cbbPalette[1], cbbPalette[2], cbbPalette[3], cbbPalette[5], cbbPalette[8], cbbPalette[6], cbbPalette[7], "purple"), name = " ") +
  scale_shape_manual(values = c(21, 22, 23, 25, 24), label = c("Part of Core \n Non-recovered or Indirect", "Part of Core \n Recovered", "Not part of core \n Local extinction", "Not part of core \n Non-recovered or Indirect", "Not part of core \n Recovered"), name = "") +
  theme_classic() + 
  facet_wrap(~Treatment) +
  theme(
   strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   panel.spacing = unit(2, "lines"), 
   legend.position="right", legend.box = "verticle",
   legend.key.size=unit(2,"point")) +
  guides(fill = guide_legend(override.aes=list(shape=21))) + 
  xlab("log(Mean Relative Abundance)") + 
  ylab("Occupancy") + 
  ggtitle("Soybean")
```

```{r}
Fig.2ab <- ggarrange(Fig.2A.Composition.ANCOM, Fig.2C.Soy.Recover.ANCOM, nrow = 2, ncol = 1, align = "h", labels = c("(a)", "(b)"), common.legend = T, legend = "right")

Fig.2cd <- ggarrange(Fig.2B.Soy.Trem.Composition.ANCOM, Trem.OTUs.recov, nrow = 2, ncol = 1, align = "h", labels = c("(c)", "(d)"), common.legend = T, legend = "right")

Fig.2efg <- ggarrange(soybean.bull.richness, corn.bull.richness, abund.occ.bull.soy, nrow = 3, ncol = 1, align = "h", labels = c("(e)", "(f)", "(g)"), common.legend = F)

Fig.2abcd <- ggarrange(Fig.2ab, Fig.2cd, ncol = 1, nrow = 2)

Fig.2 <- ggarrange(Fig.2abcd, Fig.2efg, ncol = 2, nrow = 1)
```

Supplemental tables
- Table with the OTU that was significantly different, and part of core, and recovered or not with the occupancy and relative abundance at R3
```{r}
write.csv(soy.classification.combined, "Soybean_supplementaltable.csv")
write.csv(corn.suptable, "Corn_supplementaltable.csv")
write.csv(prokaryote.cores, "prokaryote_core.csv")
write.csv(fungi.cores, "fungi_core.csv")
```
