---
title: "Networks"
author: "Zachary Noel"
date: "6/3/2020"
output: html_document
---

```{r SOURCE}
source('functions_themes.R')
```

```{r setup, include=FALSE}
#source('http://bioconductor.org/biocLite.R')
#biocLite('phyloseq')
library(phyloseq)
library(decontam)
#browseVignettes("phyloseq")
library(devtools)
#install_github("zdk123/SpiecEasi")
library(SpiecEasi)
packages <- c("biomformat", "qiimer", "MASS", "ape", "ggplot2", "plyr", "indicspecies", "labdsv", "dplyr", "reshape", "vegan", "metacoder", "lme4", "lsmeans", "ggpmisc", "ggpubr", "tidyverse", "doParallel", "DT", "exactRankTests", "foreach", "Rcpp", "shiny", "coin", "igraph", "SpiecEasi", "ggsci", "RCy3", "ggrepel", "cowplot", "ggalt", "ggfortify", "emmeans", "data.table", "randomForest", "rfUtilities", "caret", "tidyr")
ipak(packages)

packageVersion("vegan")
packageVersion('phyloseq')

#install.packages("remotes")
#remotes::install_github("DanielSprockett/reltools")
library(reltools)
#install.packages("minpack.lm")
library(minpack.lm)
#Models for the whole community
#install.packages("devtools")
library(devtools)
#install_github("DanielSprockett/tyRa")
#install.packages("Hmisc")
library(Hmisc)
library(Biostrings)
library(ggforce)
library(cowplot)
library(randomForest)
library(rfUtilities) # to test model significance
library(caret) # to get leave-one-out cross-validation accuracies and also contains the nearZeroVar function 
library(NetworkExtinction)
library(ggraph)
library(tidygraph)
library(vegan)
#remotes::install_github("schuyler-smith/phyloschuyler")
library(phylosmith)
```

## FUNGI CSS NORM and Non-normalized RDS
```{r FUNGI CSS NORM - RDS}
fungi.no.norm.filt <- readRDS(file = "Preprocessing/Fungicide_Fungi_RootsAndLeaves_1000readsmin_nonorm_022221.rds")
fungi.CSS.norm.filt <- readRDS(file = "Preprocessing/Fungicide_Fungi_RootsAndLeaves_1000seqMin_CSSNorm_022221.rds")
```

## PROKARYOTE CSS NORM and Non-normalized RDS
```{r}
# Restore the object
bacteria.no.norm.filt <- readRDS("Preprocessing/Fungicide_Bacteria_RootsAndLeaves_newtaxa_nonorm_010621.rds")
bacteria.css.norm <- readRDS(file = "Preprocessing/Fungicide_Bacteria_RootsAndLeaves_newtaxa_CSSnorm_010621.rds")
```

# NETWORKS 
- The input data for these networks is the fungi.no.norm or the prok.no.norm
- These are datasets with at least 1000 reads, most soybean leaf samples passed this filter 
- samples that did not sequence well enough to be included and need to be dropped for this analysis = 
      T2R6FBR3L in fungal database (R3)
      T2R1FCR6L in prokaryote database (R6)
      
- Spiec Easi does its own normalization procedure so the input is raw counts 
- However we will have to take out rare taxa with large variation 
- I am going to make a large network and pull out only the OTUs present in samples I want to analyze 
- This meta-network approach was done in Yang et al. 2019
## META-NETWORK

### Networks based on splitting each growth stage fungicide combo, then subgraphs by management

## One big meta-network Soybean
-has to be done on the HPCC - I am doing the same filtering as the ANCOM analysis, (i.e., mean RA > 1e-5 and with OTUs that had less than 5% occupancy)
```{r}
fungi.no.norm <- phylosmith::set_sample_order(fungi.no.norm.filt, "SampleID")
bacteria.no.norm <- phylosmith::set_sample_order(bacteria.no.norm.filt, "SampleID")

abund_occ <- function(otu.table) {
  otu_PA <- 1*((otu.table>0)==1)  # presence-absence data
otu_occ <- rowSums(otu_PA)/ncol(otu_PA) # Occupancy

otu_rel <- apply(decostand(otu.table, method="total", MARGIN=2),1, mean)     # mean relative abundance
occ_abun <- add_rownames(as.data.frame(cbind(otu_occ, otu_rel)),'otu') # combining occupancy and abundance data frame
return(occ_abun)
}

##### getting fOTUs with mean RA >= 1e-5 (0.00001) and occupancy >= 5% across all soy leaf samples  ######

# filtering the samples to soybean leaf samples
soy.leaf.fungi.keep <- fungi.no.norm %>%
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & SampleID != "T2R6FBR3L") %>%
  otu_table() %>% # selecting the OTU table 
  as("matrix") %>% # turning the OTU table into a matrix for the function
  abund_occ() %>% # function defined above
  subset(otu_occ >= 0.05 & otu_rel >= 1e-5) %>% # filtering
  dplyr::select(otu) %>%
  as.matrix() %>%
  as.character()

# Take out the sample that is not present in the bacterial library 
fungi_leaf_soy <- fungi.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & SampleID != "T2R6FBR3L") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) 

# Subset Samples 
fungi_leaf_soy2 <- prune_taxa(soy.leaf.fungi.keep, fungi_leaf_soy) # remove the taxa below 1e-5 and less than 5% occupancy

saveRDS(fungi_leaf_soy, "NetworkRDS/Input/fungi_leaf_soy.RDS")

##### getting pOTUs with mean RA >= 1e-5 (0.00001) and occupancy >= 5% across all soy leaf samples  ######

# filtering the samples to soybean leaf samples
soy.leaf.prok.keep <- bacteria.no.norm %>%
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & SampleID != "T2R6FBR3L") %>%
  otu_table() %>% # selecting the OTU table 
  as("matrix") %>% # turning the OTU table into a matrix for the function
  abund_occ() %>% # function defined above
  subset(otu_occ >= 0.05 & otu_rel >= 1e-5) %>% # filtering
  dplyr::select(otu) %>%
  as.matrix() %>%
  as.character()

prok_leaf_soy <- bacteria.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Soy" & SampleID != "T2R1FCR6L") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) 

# Subset Samples 
prok_leaf_soy2 <- prune_taxa(soy.leaf.prok.keep, prok_leaf_soy) # remove the taxa below 10-5 
saveRDS(prok_leaf_soy, "NetworkRDS/Input/prok_leaf_soy.RDS")

##### fungi Subset OTUs with mean RA less than 1-5 corn  ######
fungi.samples.take.out.corn <- c("Corn2017LeafObjective2Collection1T1R1CBA3", "Corn2017LeafObjective2Collection1T2R1FBE6",
"Corn2017LeafObjective2Collection1T2R2FAA7", "Corn2017LeafObjective2Collection2T2R6FCE2",
"Corn2017LeafObjective2Collection3T1R5FCB5")

corn.leaf.fungi.keep <- fungi.no.norm %>%
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & SampleID %!in% fungi.samples.take.out.corn) %>%
  otu_table() %>% # selecting the OTU table 
  as("matrix") %>% # turning the OTU table into a matrix for the function
  abund_occ() %>% # function defined above
  subset(otu_occ >= 0.05 & otu_rel >= 1e-5) %>% # filtering
  dplyr::select(otu) %>%
  as.matrix() %>%
  as.character()

# Take out the sample that is not present in the bacterial library 
fungi_leaf_corn <- fungi.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & SampleID %!in% fungi.samples.take.out.corn) %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) 

# Subset Samples 
fungi_leaf_corn2 <- prune_taxa(corn.leaf.fungi.keep, fungi_leaf_corn) # remove the taxa below 10-5 


##### bacteria Subset OTUs with mean RA less than 1-5 corn ######
corn.leaf.prok.keep <- bacteria.no.norm %>%
  subset_samples(Compartment == "Leaf" & Crop == "Corn" & SampleID %!in% fungi.samples.take.out.corn) %>%
  otu_table() %>% # selecting the OTU table 
  as("matrix") %>% # turning the OTU table into a matrix for the function
  abund_occ() %>% # function defined above
  subset(otu_occ >= 0.05 & otu_rel >= 1e-5) %>% # filtering
  dplyr::select(otu) %>%
  as.matrix() %>%
  as.character()

prok_leaf_corn <- bacteria.no.norm %>% 
  subset_samples(Compartment == "Leaf" & Crop == "Corn") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) 

# Subset Samples 
prok_leaf_corn2 <- prune_taxa(corn.leaf.prok.keep, prok_leaf_corn) # remove the taxa below 10-5 

all.equal(sample_names(fungi_leaf_corn), sample_names(prok_leaf_corn))

saveRDS(fungi_leaf_corn, "NetworkRDS/Input/fungi_leaf_corn.RDS")
saveRDS(prok_leaf_corn, "NetworkRDS/Input/prok_leaf_corn.RDS")
```

- This is the code that was run on the high-performance computer. I would not recommend you run this code locally... 
```{r}
spiec.easi.network <- function(prok.phyloseq, fungi.phyloseq){
  set.seed(12324)
spiec_output <- spiec.easi(list(prok.phyloseq, fungi.phyloseq), method='mb', lambda.min.ratio=1e-2, nlambda=100, pulsar.params=list(thresh = 0.05, seed = 10020))
stab <- getStability(spiec_output) # 0.0479 - good
soy_net_weights <- symBeta(getOptBeta(spiec_output), mode="maxabs")
weights <- Matrix::summary(t(soy_net_weights))[,3]
names_taxa_net <- c(taxa_names(prok.phyloseq), taxa_names(fungi.phyloseq))
nat.con <- pulsar::natural.connectivity(getRefit(spiec_output))
graph.disimilarity <- as.matrix(pulsar::graph.diss(getRefit(spiec_output)))
spiec_output_graph <- adj2igraph(getRefit(spiec_output),
                                    edge.attr = list(weight=weights),
                                    vertex.attr = list(name=names_taxa_net))

degree.distribution <- data.frame(degree(spiec_output_graph)); colnames(degree.distribution) <- "degree"
#degree.distribution$GrowthStage <- as.character(GrowthStage)
#degree.distribution$Fungicide <- as.character(Fungicide)

return.list <- list(igraph.object = spiec_output_graph, stability = stab, degree_dist = degree.distribution, natural.connect = nat.con, graph.disim = graph.disimilarity)
return(return.list)
}

#meta.net <- spiec.easi.network(prok_leaf_soy, fungi_leaf_soy)
#meta.net$stability
```

## Load in the Meta-Network objects
- Load in the meta.net object from the hpcc
```{r}
meta.net.soy <- readRDS("NetworkRDS/Output/soy_leaf_network_022421.rds")
meta.net.soy$igraph.object

meta.net.corn <- readRDS("NetworkRDS/Output/corn_leaf_network_022421.rds")
meta.net.corn$igraph.object
```

## Loading in Metadata associated with each OTU such as, taxonomy, differential abundance, core membership, relative abundance etc. 
FULL Taxonomy lists
```{r}
fungi.taxonomy <- fungi.no.norm %>%
  tax_table() %>%
  as("matrix") %>%
  tibble::as_tibble() %>%
  dplyr::select(OTU_ID, Kingdom, Phylum, Class, Order, Family, Genus, BestMatch, Taxonomy)
bacteria.taxonomy <- bacteria.no.norm %>%
  tax_table() %>%
  as("matrix") %>%
  tibble::as_tibble() %>%
  dplyr::select(OTU_ID, Kingdom, Phylum, Class, Order, Family, Genus, BestMatch, Taxonomy)

taxonomy.combined <- rbind.data.frame(fungi.taxonomy, bacteria.taxonomy)
```

-Relative abundance of each OTU
```{r}
#### getting the mean relative abundance of each fungal OTU ####
fungi_obj2.ra <- fungi.no.norm %>%   
  subset_samples(Compartment == "Leaf" & Crop == "Soy") %>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt.fast()                                      # Melt to long format
mean.rel.abund.fungi <- fungi_obj2.ra %>% 
  group_by(OTU) %>% 
  nest() %>%
  mutate(Kingdom = purrr::map(data, ~.x$Kingdom[[1]])) %>%
  mutate(Phylum = purrr::map(data, ~.x$Phylum[[1]])) %>%
  mutate(Class = purrr::map(data, ~.x$Class[[1]])) %>%
  mutate(Order = purrr::map(data, ~.x$Order[[1]])) %>%
  mutate(Family = purrr::map(data, ~.x$Family[[1]])) %>%
  mutate(Genus = purrr::map(data, ~.x$Genus[[1]])) %>%
  mutate(BestMatch = purrr::map(data, ~.x$BestMatch[[1]])) %>%
  mutate(Taxonomy = purrr::map(data, ~.x$Taxonomy[[1]])) %>%
  mutate(mean.relabund = purrr::map(data,~mean(.$Abundance*100))) %>%
  mutate(SE.relabund = purrr::map(data,~sd(.$Abundance*100)/sqrt(length(.$Abundance*100)))) %>%
  unnest(mean.relabund, SE.relabund, Kingdom, Phylum, Class, Order, Family, Genus, BestMatch, Taxonomy) %>%
  dplyr::select(OTU, mean.relabund, SE.relabund, Kingdom, Phylum, Class, Order, Family, Genus, BestMatch, Taxonomy)

#### getting the mean relative abundance of each bacterial OTU ####
prok_obj2.ra <- bacteria.no.norm %>%   
  subset_samples(Compartment == "Leaf" & Crop == "Soy") %>%
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt.fast()                                      # Melt to long format
mean.rel.abund.prok <- prok_obj2.ra %>% 
  group_by(OTU) %>% 
  nest() %>%
   mutate(Kingdom = purrr::map(data, ~.x$Kingdom[[1]])) %>%
  mutate(Phylum = purrr::map(data, ~.x$Phylum[[1]])) %>%
  mutate(Class = purrr::map(data, ~.x$Class[[1]])) %>%
  mutate(Order = purrr::map(data, ~.x$Order[[1]])) %>%
  mutate(Family = purrr::map(data, ~.x$Family[[1]])) %>%
  mutate(Genus = purrr::map(data, ~.x$Genus[[1]])) %>%
  mutate(BestMatch = purrr::map(data, ~.x$BestMatch[[1]])) %>%
  mutate(Taxonomy = purrr::map(data, ~.x$Taxonomy[[1]])) %>%
  mutate(mean.relabund = purrr::map(data,~mean(.$Abundance*100))) %>%
  mutate(SE.relabund = purrr::map(data,~sd(.$Abundance*100)/sqrt(length(.$Abundance*100)))) %>%
  unnest(mean.relabund, SE.relabund, Kingdom, Phylum, Class, Order, Family, Genus, BestMatch, Taxonomy) %>%
  dplyr::select(OTU, mean.relabund, SE.relabund, Kingdom, Phylum, Class, Order, Family, Genus, BestMatch, Taxonomy)

#### Join the abundances together ####
mean.rel.abund.combined <- rbind.data.frame(mean.rel.abund.prok, mean.rel.abund.fungi)
```

### Network Complexity 

Calculate network complexity and architecture based on each subgraph containing only the OTUs present in that sample. 

Corn
```{r}
##### Network Complexity Stats Corn #####
# Going to loop through all the samples and get the number of edges and nodes for each sub-graph

### INPUT ###
network.object <- meta.net.corn$igraph.object
phyloseq.object <- fungi_leaf_corn # it doesn't really matter if you pick the fungal or prokaryote as long as they have the same samples
network.complexity2 <- NULL
edge.complexity <- NULL
node.stats <- NULL

sample.levels <- sample_data(phyloseq.object)$SampleID
for (i in seq_along(sample.levels)) {
fungi.soy <- fungi.no.norm %>%
  subset_samples(SampleID %in% sample.levels[[i]]) %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  otu_table() %>%
  rownames()

prok.soy <- bacteria.no.norm %>%
  subset_samples(SampleID %in% sample.levels[[i]]) %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  otu_table() %>%
  rownames()

sample.data <- fungi.no.norm %>%
  subset_samples(SampleID %in% sample.levels[[i]]) %>%
  sample_data() %>%
  as.tibble() %>%
  as.data.frame()

samples <- sample.data %>%
  dplyr::select(SampleID, Crop, DateSampled, GrowthStage, Treatment, Rep, Sample, Fungicide)

subset.taxa <- c(fungi.soy, prok.soy)
print(paste("N taxa present in", sample.levels[[i]]))
print(length(subset.taxa))
###### Subsetting the network
network_dc_switch <- induced_subgraph(network.object, which(V(network.object)$name %in% subset.taxa))
print(network_dc_switch)
# Delete isolated nodes
#network_dc_switch <- delete.vertices(network_dc_switch, which(degree(network_dc_switch, mode = "all")== 0))
network_dc_switch_no_weight <- delete_edge_attr(network_dc_switch, "weight")
#### Node stats ####
wtc <- cluster_walktrap(network_dc_switch_no_weight)
trans <- igraph::transitivity(network_dc_switch_no_weight)
cfg <- cluster_fast_greedy(network_dc_switch_no_weight)
mod <- modularity(network_dc_switch_no_weight, membership(wtc))
mod2 <- modularity(network_dc_switch_no_weight, membership(cfg))
connect <- pulsar::natural.connectivity(as_adj(network_dc_switch_no_weight), norm = TRUE)
print(mod)
hubs <- data.frame(hub_score(network_dc_switch_no_weight)$vector); colnames(hubs) <- c("hubscore")
hubs$OTU <- V(network_dc_switch_no_weight)$name
hubs$betweeness <- betweenness(network_dc_switch_no_weight, directed = FALSE, weights = NULL) 
hubs$pr <- page_rank(network_dc_switch_no_weight)$vector 
hubs$degree <- degree(network_dc_switch_no_weight)
hubs$module <- wtc$membership

hubs <- merge(hubs, samples)
nodes.stats <- hubs
node.stats2 <- left_join(nodes.stats, mean.rel.abund.combined, by = "OTU")
node.stats <- rbind.data.frame(node.stats, node.stats2)
########

###### Network Complexity ######
n.nodes <- length(V(network_dc_switch)) # N nodes 
n.edges <- length(E(network_dc_switch)) # N edges
Sample <- sample.levels[[i]]
############

###### Edge Stats #######
edge.network <- data.frame(ends(network_dc_switch, E(network_dc_switch))) 
colnames(edge.network) <- c("from", "to")

edge.network <- left_join(edge.network, taxonomy.combined, by = c("from" = "OTU_ID"))

colnames(edge.network) <- c("from", "to", "Kingdom_from", "Phylum_from", "Class_from", "Order_from", 
                            "Family_from", "Genus_from", "Label_from", "Species_from")

edge.network <- left_join(edge.network, taxonomy.combined, by = c("to" = "OTU_ID"))
colnames(edge.network) <- c("from", "to", "Kingdom_from", "Phylum_from", "Class_from", "Order_from", 
                            "Family_from", "Genus_from", "Label_from", "Species_from", 
                            "Kingdom_to", "Phylum_to", "Class_to", "Order_to", 
                            "Family_to", "Genus_to", "Label_to", "Species_to")
edge.network$weights <- E(network_dc_switch)$weight
edge.network$direction <- ifelse(edge.network$weights > 0, "Positive", "Negative")
edge.network$inter <- ifelse(edge.network$Kingdom_from == edge.network$Kingdom_to, "Intrakingdom", "Interkingdom")
edge.network$interXdirection <- interaction(edge.network$inter, edge.network$direction)
edge.network$Kingdom_Kingdom <- ifelse(edge.network$inter == "Intrakingdom" & edge.network$Kingdom_to == "Fungi", "Fungal-Fungal", "Prokaryote-Prokaryote")

edge.network_i <- merge(edge.network, samples)
edge.complexity <- rbind.data.frame(edge.network_i, edge.complexity)
######

network_complexity <- data.frame(phyloseq.object@sam_data[phyloseq.object@sam_data$SampleID == Sample,]) 
network_complexity_i <- cbind.data.frame(n.nodes, n.edges, mod, mod2, trans, connect, network_complexity)
network.complexity2 <- rbind.data.frame(network.complexity2, network_complexity_i)
}

### OTUPUT ###
Network.complexity.corn <- network.complexity2 
Network.edges.corn <- edge.complexity
Network.nodes.corn <- node.stats
##################################
```

Soybean
```{r}
##### Network Complexity Stats Soy #####
# Going to loop through all the samples and get the number of edges and nodes for each sub-graph

### INPUT ###
network.object <- meta.net.soy$igraph.object
phyloseq.object <- fungi_leaf_soy # it doesn't really matter if you pick the fungal or prokaryote as long as they have the same samples
network.complexity2 <- NULL
edge.complexity <- NULL
node.stats <- NULL

sample.levels <- sample_data(phyloseq.object)$SampleID
for (i in seq_along(sample.levels)) {
fungi.soy <- fungi.no.norm %>%
  subset_samples(SampleID %in% sample.levels[[i]]) %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  otu_table() %>%
  rownames()

prok.soy <- bacteria.no.norm %>%
  subset_samples(SampleID %in% sample.levels[[i]]) %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) %>%
  otu_table() %>%
  rownames()

sample.data <- fungi.no.norm %>%
  subset_samples(SampleID %in% sample.levels[[i]]) %>%
  sample_data() %>%
  as.tibble() %>%
  as.data.frame()

samples <- sample.data %>%
  dplyr::select(SampleID, Crop, DateSampled, GrowthStage, Treatment, Rep, Sample, Fungicide)

subset.taxa <- c(fungi.soy, prok.soy)
print(paste("N taxa present in", sample.levels[[i]]))
print(length(subset.taxa))
###### Subsetting the network
network_dc_switch <- induced_subgraph(network.object, which(V(network.object)$name %in% subset.taxa))
print(network_dc_switch)
# Delete isolated nodes
#network_dc_switch <- delete.vertices(network_dc_switch, which(degree(network_dc_switch, mode = "all")== 0))
network_dc_switch_no_weight <- delete_edge_attr(network_dc_switch, "weight")
#### Node stats ####
wtc <- cluster_walktrap(network_dc_switch_no_weight)
trans <- igraph::transitivity(network_dc_switch_no_weight)
cfg <- cluster_fast_greedy(network_dc_switch_no_weight)
mod <- modularity(network_dc_switch_no_weight, membership(wtc))
mod2 <- modularity(network_dc_switch_no_weight, membership(cfg))
connect <- pulsar::natural.connectivity(as_adj(network_dc_switch_no_weight), norm = TRUE)
print(mod)
hubs <- data.frame(hub_score(network_dc_switch_no_weight)$vector); colnames(hubs) <- c("hubscore")
hubs$OTU <- V(network_dc_switch_no_weight)$name
hubs$betweeness <- betweenness(network_dc_switch_no_weight, directed = FALSE, weights = NULL) 
hubs$pr <- page_rank(network_dc_switch_no_weight)$vector 
hubs$degree <- degree(network_dc_switch_no_weight)
hubs$module <- wtc$membership

hubs <- merge(hubs, samples)
nodes.stats <- hubs
node.stats2 <- left_join(nodes.stats, mean.rel.abund.combined, by = "OTU")
node.stats <- rbind.data.frame(node.stats, node.stats2)
########

###### Network Complexity ######
n.nodes <- length(V(network_dc_switch)) # N nodes 
n.edges <- length(E(network_dc_switch)) # N edges
Sample <- sample.levels[[i]]
############

###### Edge Stats #######
edge.network <- data.frame(ends(network_dc_switch, E(network_dc_switch))) 
colnames(edge.network) <- c("from", "to")

edge.network <- left_join(edge.network, taxonomy.combined, by = c("from" = "OTU_ID"))

colnames(edge.network) <- c("from", "to", "Kingdom_from", "Phylum_from", "Class_from", "Order_from", 
                            "Family_from", "Genus_from", "Label_from", "Species_from")

edge.network <- left_join(edge.network, taxonomy.combined, by = c("to" = "OTU_ID"))
colnames(edge.network) <- c("from", "to", "Kingdom_from", "Phylum_from", "Class_from", "Order_from", 
                            "Family_from", "Genus_from", "Label_from", "Species_from", 
                            "Kingdom_to", "Phylum_to", "Class_to", "Order_to", 
                            "Family_to", "Genus_to", "Label_to", "Species_to")
edge.network$weights <- E(network_dc_switch)$weight
edge.network$direction <- ifelse(edge.network$weights > 0, "Positive", "Negative")
edge.network$inter <- ifelse(edge.network$Kingdom_from == edge.network$Kingdom_to, "Intrakingdom", "Interkingdom")
edge.network$interXdirection <- interaction(edge.network$inter, edge.network$direction)
edge.network$Kingdom_Kingdom <- ifelse(edge.network$inter == "Intrakingdom" & edge.network$Kingdom_to == "Fungi", "Fungal-Fungal", "Prokaryote-Prokaryote")

edge.network_i <- merge(edge.network, samples)
edge.complexity <- rbind.data.frame(edge.network_i, edge.complexity)
######

network_complexity <- data.frame(phyloseq.object@sam_data[phyloseq.object@sam_data$SampleID == Sample,]) 
network_complexity_i <- cbind.data.frame(n.nodes, n.edges, mod, mod2, trans, connect, network_complexity)
network.complexity2 <- rbind.data.frame(network.complexity2, network_complexity_i)
}

### OTUPUT ###
Network.complexity.soy <- network.complexity2 
Network.edges.soy <- edge.complexity
Network.nodes.soy <- node.stats
##################################
```

Put both metanetwork objects together
```{r}
Network.complexity <- rbind.data.frame(Network.complexity.corn, Network.complexity.soy)
Network.nodes <- rbind.data.frame(Network.nodes.corn, Network.nodes.soy)
Network.edges <- rbind.data.frame(Network.edges.soy, Network.edges.corn)
```

```{r}
# Complexity = edges per node
Network.complexity$complexity <- Network.complexity$n.edges/Network.complexity$n.nodes 

# Facet Labels for plots
Network.complexity$Treatment_label <- ifelse(Network.complexity$Treatment == "T1", "Conv.", "No-till")
Network.nodes$Treatment_label <- ifelse(Network.nodes$Treatment == "T1", "Conv.", "No-till")
Network.edges$Treatment_label <- ifelse(Network.edges$Treatment == "T1", "Conv.", "No-till")
```

```{r}
# Changing the date format and putting the categorical dpf 
Network.complexity$DateSampled <- as.Date(Network.complexity$DateSampled, "%d-%b-%y")
Network.complexity$DPF <- ifelse(Network.complexity$DateSampled == "2018-08-03", "0-dpf",
                                 ifelse(Network.complexity$DateSampled == "2017-06-26", "0-dpf",
                                        ifelse(Network.complexity$DateSampled == "2018-08-16", "13-dpf",
                                               ifelse(Network.complexity$DateSampled == "2017-07-5", "9-dpf",
                                                      ifelse(Network.complexity$DateSampled == "2017-07-31", "34-dpf", "33-dpf")))))
Network.complexity$DPF <- factor(Network.complexity$DPF, levels = c("0-dpf", "9-dpf", "13-dpf", "33-dpf", "34-dpf"))

Network.nodes$DateSampled <- as.Date(Network.nodes$DateSampled, "%d-%b-%y")
Network.nodes$DPF <- ifelse(Network.nodes$DateSampled == "2018-08-03", "0-dpf",
                                 ifelse(Network.nodes$DateSampled == "2017-06-26", "0-dpf",
                                        ifelse(Network.nodes$DateSampled == "2018-08-16", "13-dpf",
                                               ifelse(Network.nodes$DateSampled == "2017-07-5", "9-dpf",
                                                      ifelse(Network.nodes$DateSampled == "2017-07-31", "34-dpf", "33-dpf")))))
Network.nodes$DPF <- factor(Network.nodes$DPF, levels = c("0-dpf", "9-dpf", "13-dpf", "33-dpf", "34-dpf"))

Network.edges$DateSampled <- as.Date(Network.edges$DateSampled, "%d-%b-%y")
Network.edges$DPF <- ifelse(Network.edges$DateSampled == "2018-08-03", "0-dpf",
                                 ifelse(Network.edges$DateSampled == "2017-06-26", "0-dpf",
                                        ifelse(Network.edges$DateSampled == "2018-08-16", "13-dpf",
                                               ifelse(Network.edges$DateSampled == "2017-07-5", "9-dpf",
                                                      ifelse(Network.edges$DateSampled == "2017-07-31", "34-dpf", "33-dpf")))))
Network.edges$DPF <- factor(Network.edges$DPF, levels = c("0-dpf", "9-dpf", "13-dpf", "33-dpf", "34-dpf"))
```

Lets plot these results
```{r}
#### NUMBER OF EDGES PER NODE (i.e., Network Complexity) Soybean ####
complexity.box <- ggplot(Network.complexity[Network.complexity$Crop == "Soy",], aes(x = DPF, y = complexity, fill = Fungicide)) + 
  geom_boxplot(outlier.shape = NA) + 
  geom_point(position = position_jitterdodge(), shape = 21) +
  ylab("Linkage density") + 
  xlab("")+
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = T, label.y = 7.5, size = 6) +  
  scale_fill_manual(values=c(cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide"), name = "") +
  theme_classic() +
  labs(color = "") + 
  ylim(c(2, 8.5)) +
  facet_wrap(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top")

#### NUMBER OF NODEs Soybean####
nodes.box <- ggplot(Network.complexity[Network.complexity$Crop == "Soy",], aes(x = DPF, y = n.nodes, fill = Fungicide)) + 
  geom_boxplot(outlier.shape = NA) + 
  geom_point(position = position_jitterdodge(), shape =21) +
  ylab("Nodes") + 
  xlab("")+
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = T, label.y = 550, size = 6) +  
  scale_fill_manual(values=c(cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide")) +
  theme_classic() +
  labs(color = "") + 
  ylim(c(100, 600)) +
  facet_wrap(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top")

#### NUMBER OF EDGES PER NODE (i.e., Network Complexity) Corn ####
complexity.box.corn <- ggplot(Network.complexity[Network.complexity$Crop == "Corn",], aes(x = DPF, y = complexity, color = Fungicide)) + 
  geom_boxplot(outlier.shape = NA) + 
  geom_point(position = position_jitterdodge()) +
  ylab("Linkage density") + 
  xlab("")+
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = T, label.y = 1, size = 6) +  
  scale_color_manual(values=c(cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide")) +
  theme_classic() +
  labs(color = "") + 
  facet_wrap(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top")

#### NUMBER OF NODEs Corn####
nodes.box.corn <- ggplot(Network.complexity[Network.complexity$Crop == "Corn",], aes(x = DPF, y = n.nodes, color = Fungicide)) + 
  geom_boxplot(outlier.shape = NA) + 
  geom_point(position = position_jitterdodge()) +
  ylab("Nodes") + 
  xlab("")+
  stat_compare_means(method = "wilcox", aes(label = ..p.adj..), hide.ns = T, label.y = 100, size = 6) +  
  scale_color_manual(values=c(cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide")) +
  theme_classic() +
  labs(color = "") + 
  facet_wrap(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top")


ggpubr::ggarrange(complexity.box, nodes.box, nrow = 1, labels = c("(a)", "(b)"), common.legend = T)
```

```{r}
# cumulative positive Edges between bulleribasidiaceaea and bacteria in non-disturbed plots #Phylum level
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Kingdom_from == "Bacteria" & Fungicide == "C" & direction == "Positive") %>%
  mutate(Phylum_from2 = replace_na(Phylum_from, "unidentified")) %>%
  dplyr::group_by(Phylum_from2) %>%
  dplyr::tally() %>%
  ungroup() %>%
    complete(Phylum_from2, fill = list(n = 0)) %>%
  mutate(freq = n / sum(n)) %>%
  arrange(-freq)

# positive Edges between bulleribasidiaceaea and bacteria in non-disturbed plots Class level
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Kingdom_from == "Bacteria" & Fungicide == "C" & direction == "Positive") %>%
  dplyr::mutate(Class_from2 = replace_na(Class_from, "unidentified")) %>%
  dplyr::group_by(Class_from2, Fungicide) %>%
  dplyr::tally() %>%
  ungroup() %>%
  complete(Class_from2, fill = list(n = 0)) %>%
  mutate(freq = n / sum(n)) %>%
  arrange(-freq)

# positive Edges between bulleribasidiaceaea and bacteria in non-disturbed plots Family level
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Kingdom_from == "Bacteria" & Fungicide == "C" & direction == "Positive" & Crop == "Soy") %>%
  dplyr::mutate(Family_from2 = replace_na(Family_from, "unidentified")) %>%
  dplyr::group_by(Family_from2) %>%
  dplyr::tally() %>%
  ungroup() %>%
  complete(Family_from2, fill = list(n = 0)) %>%
  mutate(freq = n / sum(n)) %>%
  arrange(-freq)

Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Kingdom_from == "Bacteria" & Fungicide == "C" & direction == "Positive" & Crop == "Soy") %>%
  dplyr::mutate(Genus_from2 = replace_na(Genus_from, "unidentified")) %>%
  dplyr::group_by(Genus_from2) %>%
  dplyr::tally() %>%
  ungroup() %>%
  complete(Genus_from2, fill = list(n = 0)) %>%
  mutate(freq = n / sum(n)) %>%
  arrange(-freq)

Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Crop == "Soy" & DPF == "13-dpf") %>%
  subset(Kingdom_from == "Bacteria") %>%
  #subset(weights > 0.01 | weights < -0.01) %>%
  dplyr::mutate(Genus_from2 = replace_na(Genus_from, "unidentified")) %>%
  dplyr::group_by(Genus_from2, Fungicide, direction) %>%
  dplyr::tally() %>%
  ungroup() %>%
  complete(Genus_from2, fill = list(n = 0)) %>%
  mutate(freq = n / sum(n)) %>%
  arrange(-freq) %>%
  pivot_wider(id_cols = c(Genus_from2, Fungicide), names_from = c(direction), values_from = c(n)) %>%
  mutate_if(is.numeric , replace_na, replace = 0) %>%
  mutate(diff = Positive - Negative) %>%
  #mutate(Genus_label = ifelse(abs(diff) < 10, "Other", Genus_from2)) %>%
  #mutate(freq2 = ifelse(direction == "Negative", freq*-1, freq)) %>%
  ggplot(aes(x = reorder(Genus_from2, diff), y = diff, fill = Fungicide)) + 
  geom_col() +
  coord_flip() + 
  xlab("") + 
  ylab("Edge Composition") + 
  scale_y_continuous(labels = scales::percent) + 
  scale_fill_manual(values=cbbPalette) + 
  theme_classic() +
  geom_hline(yintercept = 0, linetype = "dashed", size = 1)
```

```{r}
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Family_from %in% c("Sphingomonadaceae", "Oxalobacteraceae", "Hymenobacteraceae") & direction == "Positive" & Crop == "Soy") %>%
  mutate(Family_from2 = replace_na(Family_from, "unidentified")) %>%
  ggplot() +
  geom_bar(aes(x = interaction(Fungicide, DPF), fill = Species_from), position = "fill") + 
  facet_wrap(~Treatment_label, scales = "free")
```

```{r}
positive.edges.prok <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Class_from %in% c("Alphaproteobacteria", "Actinobacteria", "Gammaproteobacteria", "Bacteroidia") & direction == "Positive") %>%
  dplyr::group_by(SampleID, Class_from) %>%
  dplyr::summarize(N = n()) %>%
  dplyr::ungroup() %>%
  complete(SampleID, nesting(Class_from), fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") 

means.prok.pos <- ggpubr::compare_means(N ~ Fungicide, data = positive.edges.prok, group.by = c("Class_from", "Crop", "Treatment_label", "DPF"), p.adjust.method = "fdr")

negative.edges.prok <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Class_from %in% c("Alphaproteobacteria", "Actinobacteria", "Gammaproteobacteria", "Bacteroidia") & direction == "Negative") %>%
  dplyr::group_by(SampleID, Class_from) %>%
  dplyr::summarize(N = n()) %>%
  dplyr::ungroup() %>%
  complete(SampleID, nesting(Class_from), fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") 

means.prok.neg <- ggpubr::compare_means(N ~ Fungicide, data = negative.edges.prok, group.by = c("Class_from", "Crop", "Treatment_label", "DPF"), p.adjust.method = "fdr")
```

```{r}
bull.class <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Class_from %in% c("Alphaproteobacteria", "Actinobacteria", "Gammaproteobacteria", "Bacteroidia") & direction == "Positive" & Crop == "Soy") %>%
  dplyr::group_by(SampleID, Class_from) %>%
  dplyr::summarize(N = n()) %>%
  dplyr::ungroup() %>%
  complete(SampleID, nesting(Class_from), fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  dplyr::group_by(Class_from, DPF, Treatment_label, Crop, Fungicide) %>%
  nest() %>% # nest this
  mutate(mean.edges = purrr::map(data, ~mean(.x$N))) %>% # loop over each combination and count how many samples
  unnest(c(mean.edges)) %>%
  pivot_wider(id_cols = c(Class_from, Treatment_label, DPF, Crop), names_from = c(Fungicide), values_from = c(mean.edges)) %>%
  mutate(diff.mean.edges = F-C) %>%
  left_join(means.prok.pos, by = c("Class_from", "Crop", "Treatment_label", "DPF")) %>%
  mutate(Sig = ifelse(p.adj <= 0.05, "Significant", "Not Significant")) %>%
  ggplot(aes(x = DPF, y = Class_from, color = diff.mean.edges,  size = Sig, shape = Sig)) + 
  geom_point() +
  ylab("") + 
  xlab("")+
  scale_color_gradient2(low = cbbPalette[[2]], mid = cbbPalette[[1]], high = cbbPalette[[1]], name = "Mean \nDifference") +
  scale_shape_manual(values = c(18,19), name = "") +
  scale_size_manual(values = c(4, 8), name = "") + 
  theme_classic() +
  labs(color = "") + 
  facet_grid(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
   legend.position="right") +
  ggtitle("Positive edges \nwith Bulleribasidiaceae")
bull.class 

bull.class.neg <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Class_from %in% c("Alphaproteobacteria", "Actinobacteria", "Gammaproteobacteria", "Bacteroidia") & direction == "Negative" & Crop == "Soy") %>%
  dplyr::group_by(SampleID, Class_from) %>%
  dplyr::summarize(N = n()) %>%
  dplyr::ungroup() %>%
  complete(SampleID, nesting(Class_from), fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  dplyr::group_by(Class_from, DPF, Treatment_label, Crop, Fungicide) %>%
  nest() %>% # nest this
  mutate(mean.edges = purrr::map(data, ~mean(.x$N))) %>% # loop over each combination and count how many samples
  unnest(c(mean.edges)) %>%
  pivot_wider(id_cols = c(Class_from, Treatment_label, DPF, Crop), names_from = c(Fungicide), values_from = c(mean.edges)) %>%
  mutate(diff.mean.edges = F-C) %>%
  left_join(means.prok, by = c("Class_from", "Crop", "Treatment_label", "DPF")) %>%
  mutate(Sig = ifelse(p.adj <= 0.05, "Significant", "Not Significant")) %>%
  ggplot(aes(x = DPF, y = Class_from, color = diff.mean.edges,  size = Sig, shape = Sig)) + 
  geom_point() +
  ylab("") + 
  xlab("")+
  scale_color_gradient2(low = cbbPalette[[2]], mid = cbbPalette[[1]], high = cbbPalette[[1]], name = "Mean \nDifference") +
  scale_shape_manual(values = c(18,19), name = "") +
  scale_size_manual(values = c(4, 8), name = "") + 
  theme_classic() +
  labs(color = "") + 
  facet_grid(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
   legend.position="right") +
  ggtitle("Positive edges \nwith Bulleribasidiaceae")

ggpubr::ggarrange(complexity.box, bull.class, nrow = 1, labels = c("A", "B"))
```


```{r}
edges.sphingo <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Genus_from %in% c("Sphingomonas")) %>%
  dplyr::group_by(SampleID) %>%
  dplyr::summarize(N = sum(weights)) %>%
  #dplyr::tally() %>%
  dplyr::ungroup() %>%
  complete(SampleID, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  subset(Crop == "Soy") %>%
  ggplot(aes(y = N, x = DPF, color = Fungicide)) +
  geom_boxplot() + 
  geom_point(position = position_jitterdodge()) +
  ylab("Positive \n edge count \n Sphingomonas") + 
  xlab("")+
  stat_compare_means(method = "wilcox", hide.ns = T, label.y = 1, size = 2) +  
  scale_color_manual(values=c(cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide")) +
  theme_classic() +
  labs(color = "") + 
  facet_wrap(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top") 

edges.hymenobacter <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Genus_from %in% c("Hymenobacter") & direction == "Positive") %>%
  dplyr::group_by(SampleID) %>%
  dplyr::summarize(N = sum(weights)) %>%
  #dplyr::tally() %>%
  dplyr::ungroup() %>%
  complete(SampleID, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  subset(Crop == "Soy") %>%
  ggplot(aes(y = N, x = DPF, color = Fungicide)) +
  geom_boxplot() + 
  geom_point(position = position_jitterdodge()) +
  ylab("Positive \nedge count \nHymenobacter") + 
  xlab("")+
  stat_compare_means(method = "wilcox", hide.ns = T, label.y = 8, size = 2) +  
  scale_color_manual(values=c(cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide")) +
  theme_classic() +
  labs(color = "") + 
  facet_wrap(~Crop*Treatment_label, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top") 

ggpubr::ggarrange(edges.sphingo, edges.hymenobacter, common.legend = T, nrow = 1, labels = c("E", "F"))
```


```{r}

fungi_leaf_soy %>%
  subset_samples(Fungicide == "C")

Network.edges[Network.edges$to == "BOTU_80",]
Network.edges$Family_to[Network.edges$from == "BOTU_21"]


Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Class_from %in% c("Alphaproteobacteria", "Actinobacteria", "Gammaproteobacteria", "Bacteroidia") & Crop == "Soy" & Fungicide == "C") %>%
  mutate(Order_from2 = replace_na(Order_from, "unidentified")) %>%
  dplyr::group_by(SampleID, Order_from2) %>%
  dplyr::summarize(N = sum(weights)) %>%
  dplyr::ungroup() %>%
  complete(SampleID, Order_from2, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  ggplot(aes(y = reorder(Order_from2, N), x = N, color = Treatment_label)) +
  stat_summary(fun.y=mean,geom="point") +
  stat_summary(fun.data = mean_ci, geom = "errorbar", width = 0.5) +
  ylab("") + 
  xlab("")+
  theme_classic() +
  labs(color = "") + 
  geom_vline(xintercept = 0, linetype = "dashed") +
  #facet_wrap(~, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top") 
```

```{r}
hymenobacter.potus <- Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Genus_from %in% c("Hymenobacter") & Crop == "Soy") %>%
  select(from) %>%
  unique() %>%
  as.matrix() %>% 
  as.character()

hymenobacter <- prune_taxa(hymenobacter.potus, bacteria.no.norm)
  hymenobacter@tax_table
reltools::save_fasta(hymenobacter, file = "hymenobacter.fasta", rank = "Taxonomy")
```

```{r}
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Class_from %in% c("Alphaproteobacteria", "Bacteroidia", "Actinobacteria") & Crop == "Soy") %>%
  dplyr::group_by(SampleID, Genus_from) %>%
  dplyr::summarize(N = sum(weights)) %>%
  dplyr::ungroup() %>%
  complete(SampleID, Genus_from, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  ggplot(aes(x = reorder(Genus_from, N), y = N, color = Fungicide)) +
  stat_summary(fun.y=mean,geom="point") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = T) +  
  ylab("Cumulative mean edge weight") + 
  xlab("")+
  theme_classic() +
  labs(color = "") + 
  coord_flip()+
  geom_vline(xintercept = 0, linetype = "dashed") +
  facet_wrap(~DPF) +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top") +
  geom_hline(yintercept = 0, ltype = "dashed")


Dioszegia 
Vishniacozyma
Hannaella

Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Genus_from %in% c("Hymenobacter", "Sphingomonas", "Methylobacterium-methylorubrum") & Crop == "Soy") %>%
  mutate(Genus_from = ifelse(Genus_from == "Methylobacterium-methylorubrum", "Methylobacterium sp.", Genus_from)) %>%
  mutate(Species_from = paste(from, Genus_from, sep="-")) %>%
  dplyr::group_by(SampleID, Species_from) %>%
  dplyr::summarize(N = sum(weights)) %>%
  dplyr::ungroup() %>%
  complete(SampleID, Species_from, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  ggplot(aes(x = reorder(Species_from, N), y = N, color = interaction(Fungicide, Treatment))) +
  geom_hline(yintercept = 0, linetype = "dashed", alpha = 0.5) +
    stat_summary(fun.y = mean,
               geom = "point",
               size = 1.5, alpha = 0.5) +
  stat_summary(fun.data = mean_se,
               geom = "errorbar", width = .2, alpha = 0.5) +
  scale_color_manual(values=c(cbbPalette[[1]], cbbPalette[[2]],
                     cbbPalette[[1]], cbbPalette[[2]]), labels = c("Control", "Fungicide")) +
  stat_compare_means(method = "wilcox", hide.ns = T, size = 3, label.y = 0) +  
  ylab("Cumulative mean edge weight") + 
  xlab("")+
  theme_classic() +
  labs(color = "") + 
  coord_flip()+
  facet_wrap(Treatment_label~DPF, scales = "free_x", ncol = 3) +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="none", 
   panel.spacing = unit(1.5, "lines")) 
```

```{r}
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & Family_from %in% c("Pleosporaceae", "Bulleribasidiaceae") & Crop == "Soy") %>%
  dplyr::group_by(SampleID, Species_from) %>%
  dplyr::summarize(N = sum(weights)) %>%
  dplyr::ungroup() %>%
  complete(SampleID, Species_from, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  ggplot(aes(x = reorder(Species_from, N), y = N, color = Fungicide)) +
  stat_summary(fun.y=mean,geom="point") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  stat_compare_means(method = "wilcox", aes(label = ..p.signif..), hide.ns = T) +  
  ylab("Cumulative mean edge weight") + 
  xlab("")+
  theme_classic() +
  labs(color = "") + 
  coord_flip()+
  geom_vline(xintercept = 0, linetype = "dashed") +
  facet_wrap(~DPF, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="none") 
```


```{r}
Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & from %in% c("BOTU_41", "BOTU_6451", "BOTU_22", "BOTU_80", "BOTU_21", "BOTU_5403", "BOTU_1874", "BOTU_24", "BOTU_10215") & Crop == "Soy") %>%
  mutate(Species_from2 = replace_na(Species_from, "unidentified")) %>%
  dplyr::group_by(SampleID, Species_from2) %>%
  dplyr::summarize(N = sum(weights)) %>%
  dplyr::ungroup() %>%
  complete(SampleID, Species_from2, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  ggplot(aes(y = reorder(Species_from2, N), x = N, color = Fungicide)) +
  stat_summary(fun.y=mean,geom="point") +
  stat_summary(fun.data = mean_ci, geom = "errorbar", width = 0.5) +
  ylab("") + 
  xlab("Mean cumulative edge weight")+
  theme_classic() +
  labs(color = "") + 
  geom_vline(xintercept = 0, linetype = "dashed") +
  facet_wrap(~DPF) +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top") 

Network.edges %>%
  subset(Family_to %in% c("Bulleribasidiaceae") & from %in% c("BOTU_129", "BOTU_44", "BOTU_528", "BOTU_5403", "BOTU_109", "BOTU_80", "BOTU_557", "BOTU_5895", "BOTU_483") & Crop == "Soy") %>%
  mutate(Species_from2 = replace_na(Species_from, "unidentified")) %>%
  dplyr::group_by(SampleID, Species_from2) %>%
  dplyr::summarize(N = sum(weights)) %>%
  dplyr::ungroup() %>%
  complete(SampleID, Species_from2, fill = list(N = 0)) %>%
  left_join(Network.complexity, by = "SampleID") %>%
  ggplot(aes(y = reorder(Species_from2, N), x = N, color = Fungicide)) +
  stat_summary(fun.y=mean,geom="point") +
  stat_summary(fun.data = mean_ci, geom = "errorbar", width = 0.5) +
  ylab("") + 
  xlab("")+
  theme_classic() +
  labs(color = "") + 
  geom_vline(xintercept = 0, linetype = "dashed") +
  facet_wrap(~DPF, scales = "free") +
  theme(strip.background = element_rect(color="white", fill="white", size=1.5, linetype="solid"),
   strip.text.x = element_text(size = 12, color = "black"),
   legend.position="top") 
```



